{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ATAC-Seq analysis example\n",
    "### Original authors\n",
    "Délara Sabéran-Djoneidi, Anne-Laure Schang, Kate Wooley-Allen, Sascha Ott\n",
    "### Update\n",
    "Alix Silvert, Magali Hennion\n",
    "## Intro and general description\n",
    "\n",
    "This document aims to be a detailed example of an ATAC-Seq analysis used in a research context. The goal here is to help you draft your own experiments from this particular base, by walking you through the steps we took and by pointing out the parts where the specificity of your experimental set will require some tinkering.\n",
    "\n",
    "## Clone the git repository\n",
    "\n",
    "The documents related to this tutorial are available at https://github.com/parisepigenetics/ATAC-seq. The rest of this description assumes that you have cloned this repository (using `git clone`) and that you are in `ATAC-seq` folder. The `pwd` command allows you to check where you are. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/mag/ATACseq/TestJupyterNotebook\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004h(ATAC-Seq) \u001b[?2004l\u001b[?2004l"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "## if you're not in the ATAC-seq folder, use cd to enter it.\n",
    "## cd PATHto/ATAC-seq\n",
    "cd ATAC-seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Conda environment\n",
    "\n",
    "In order to be sure to have the appropriate tools to run this analysis, we provide a Conda environment summarizing them. In order to use it, please download [Conda](https://docs.conda.io/projects/conda/en/latest/user-guide/install/download.html) and run the following line.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conda env create -f conda_environment.yml"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "All the remaining of the analysis must be run in the ATAC-seq Conda environment. You have to activate the environment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\r",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate ATAC-Seq"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Download test dataset\n",
    "\n",
    "To test this tutorial, you can download a small test dataset. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-09-24 15:42:10--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822153_1.fastq.gz\n",
      "SSL_INIT\n",
      "Certificat de l’autorité de certification « /etc/ssl/certs/ca-certificates.crt » chargé\n",
      "Résolution de raw.githubusercontent.com (raw.githubusercontent.com)… 185.199.109.133, 185.199.110.133, 185.199.111.133, ...\n",
      "Connexion à raw.githubusercontent.com (raw.githubusercontent.com)|185.199.109.133|:443… connecté.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5272547 (5,0M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822153_1.fastq.gz »\n",
      "\n",
      "SRR1822153_1.fastq. 100%[===================>]   5,03M  3,46MB/s    ds 1,5s    \n",
      "\n",
      "2021-09-24 15:42:11 (3,46 MB/s) — « data/raw_reads/SRR1822153_1.fastq.gz » sauvegardé [5272547/5272547]\n",
      "\n",
      "--2021-09-24 15:42:11--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822154_1.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5253787 (5,0M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822154_1.fastq.gz »\n",
      "\n",
      "SRR1822154_1.fastq. 100%[===================>]   5,01M  2,85MB/s    ds 1,8s    \n",
      "\n",
      "2021-09-24 15:42:14 (2,85 MB/s) — « data/raw_reads/SRR1822154_1.fastq.gz » sauvegardé [5253787/5253787]\n",
      "\n",
      "--2021-09-24 15:42:14--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822157_1.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5382195 (5,1M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822157_1.fastq.gz »\n",
      "\n",
      "SRR1822157_1.fastq. 100%[===================>]   5,13M  2,90MB/s    ds 1,8s    \n",
      "\n",
      "2021-09-24 15:42:16 (2,90 MB/s) — « data/raw_reads/SRR1822157_1.fastq.gz » sauvegardé [5382195/5382195]\n",
      "\n",
      "--2021-09-24 15:42:16--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822158_1.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5350303 (5,1M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822158_1.fastq.gz »\n",
      "\n",
      "SRR1822158_1.fastq. 100%[===================>]   5,10M  3,07MB/s    ds 1,7s    \n",
      "\n",
      "2021-09-24 15:42:18 (3,07 MB/s) — « data/raw_reads/SRR1822158_1.fastq.gz » sauvegardé [5350303/5350303]\n",
      "\n",
      "--2021-09-24 15:42:18--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822153_2.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5254354 (5,0M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822153_2.fastq.gz »\n",
      "\n",
      "SRR1822153_2.fastq. 100%[===================>]   5,01M  3,21MB/s    ds 1,6s    \n",
      "\n",
      "2021-09-24 15:42:19 (3,21 MB/s) — « data/raw_reads/SRR1822153_2.fastq.gz » sauvegardé [5254354/5254354]\n",
      "\n",
      "--2021-09-24 15:42:19--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822154_2.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5315718 (5,1M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822154_2.fastq.gz »\n",
      "\n",
      "SRR1822154_2.fastq. 100%[===================>]   5,07M  3,40MB/s    ds 1,5s    \n",
      "\n",
      "2021-09-24 15:42:21 (3,40 MB/s) — « data/raw_reads/SRR1822154_2.fastq.gz » sauvegardé [5315718/5315718]\n",
      "\n",
      "--2021-09-24 15:42:21--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822157_2.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5380339 (5,1M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822157_2.fastq.gz »\n",
      "\n",
      "SRR1822157_2.fastq. 100%[===================>]   5,13M  3,52MB/s    ds 1,5s    \n",
      "\n",
      "2021-09-24 15:42:23 (3,52 MB/s) — « data/raw_reads/SRR1822157_2.fastq.gz » sauvegardé [5380339/5380339]\n",
      "\n",
      "--2021-09-24 15:42:23--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR1822158_2.fastq.gz\n",
      "SSL_INIT\n",
      "Réutilisation de la connexion existante à raw.githubusercontent.com:443.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 5401912 (5,2M) [application/octet-stream]\n",
      "Sauvegarde en : « data/raw_reads/SRR1822158_2.fastq.gz »\n",
      "\n",
      "SRR1822158_2.fastq. 100%[===================>]   5,15M  3,77MB/s    ds 1,4s    \n",
      "\n",
      "2021-09-24 15:42:25 (3,77 MB/s) — « data/raw_reads/SRR1822158_2.fastq.gz » sauvegardé [5401912/5401912]\n",
      "\n",
      "Terminé — 2021-09-24 15:42:25 —\n",
      "Temps total effectif : 15s\n",
      "Téléchargés : 8 fichiers, 41M en 13s (3,25 MB/s)\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "wget https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR182215{3,4,7,8}_1.fastq.gz https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/testdata/SRR182215{3,4,7,8}_2.fastq.gz -P data/raw_reads/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "And the corresponding reference files. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;31mSRR1822153_1.fastq.gz\u001b[0m  \u001b[01;31mSRR1822154_2.fastq.gz\u001b[0m  \u001b[01;31mSRR1822158_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822153_2.fastq.gz\u001b[0m  \u001b[01;31mSRR1822157_1.fastq.gz\u001b[0m  \u001b[01;31mSRR1822158_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822154_1.fastq.gz\u001b[0m  \u001b[01;31mSRR1822157_2.fastq.gz\u001b[0m\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls data/raw_reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "--2021-09-24 15:42:42--  https://github.com/nf-core/test-datasets/raw/atacseq/reference/genome.fa\n",
      "SSL_INIT\n",
      "Certificat de l’autorité de certification « /etc/ssl/certs/ca-certificates.crt » chargé\n",
      "Résolution de github.com (github.com)… 140.82.121.3\n",
      "Connexion à github.com (github.com)|140.82.121.3|:443… connecté.\n",
      "requête HTTP transmise, en attente de la réponse… 302 Found\n",
      "Emplacement : https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/reference/genome.fa [suivant]\n",
      "--2021-09-24 15:42:43--  https://raw.githubusercontent.com/nf-core/test-datasets/atacseq/reference/genome.fa\n",
      "SSL_INIT\n",
      "Résolution de raw.githubusercontent.com (raw.githubusercontent.com)… 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
      "Connexion à raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443… connecté.\n",
      "requête HTTP transmise, en attente de la réponse… 200 OK\n",
      "Taille : 12359807 (12M) [text/plain]\n",
      "Sauvegarde en : « reference/genome.fa.1 »\n",
      "\n",
      "genome.fa.1         100%[===================>]  11,79M  2,66MB/s    ds 4,4s    \n",
      "\n",
      "2021-09-24 15:42:48 (2,65 MB/s) — « reference/genome.fa.1 » sauvegardé [12359807/12359807]\n",
      "\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "wget https://github.com/nf-core/test-datasets/raw/atacseq/reference/genome.fa -P reference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Initial quality control\n",
    "\n",
    "In order to see how much the pre-processing improves the data, it is a good practice to look at various statistics on the raw dataset. FastQC provides a good summary of those."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR1822153_1.fastq.gz\n",
      "Approx 5% complete for SRR1822153_1.fastq.gz\n",
      "Approx 10% complete for SRR1822153_1.fastq.gz\n",
      "Approx 15% complete for SRR1822153_1.fastq.gz\n",
      "Approx 20% complete for SRR1822153_1.fastq.gz\n",
      "Approx 25% complete for SRR1822153_1.fastq.gz\n",
      "Approx 30% complete for SRR1822153_1.fastq.gz\n",
      "Approx 35% complete for SRR1822153_1.fastq.gz\n",
      "Approx 40% complete for SRR1822153_1.fastq.gz\n",
      "Approx 45% complete for SRR1822153_1.fastq.gz\n",
      "Approx 50% complete for SRR1822153_1.fastq.gz\n",
      "Approx 55% complete for SRR1822153_1.fastq.gz\n",
      "Approx 60% complete for SRR1822153_1.fastq.gz\n",
      "Approx 65% complete for SRR1822153_1.fastq.gz\n",
      "Approx 70% complete for SRR1822153_1.fastq.gz\n",
      "Approx 75% complete for SRR1822153_1.fastq.gz\n",
      "Approx 80% complete for SRR1822153_1.fastq.gz\n",
      "Approx 85% complete for SRR1822153_1.fastq.gz\n",
      "Approx 90% complete for SRR1822153_1.fastq.gz\n",
      "Approx 95% complete for SRR1822153_1.fastq.gz\n",
      "Approx 100% complete for SRR1822153_1.fastq.gz\n",
      "Analysis complete for SRR1822153_1.fastq.gz\n",
      "Started analysis of SRR1822154_1.fastq.gz\n",
      "Approx 5% complete for SRR1822154_1.fastq.gz\n",
      "Approx 10% complete for SRR1822154_1.fastq.gz\n",
      "Approx 15% complete for SRR1822154_1.fastq.gz\n",
      "Approx 20% complete for SRR1822154_1.fastq.gz\n",
      "Approx 25% complete for SRR1822154_1.fastq.gz\n",
      "Approx 30% complete for SRR1822154_1.fastq.gz\n",
      "Approx 35% complete for SRR1822154_1.fastq.gz\n",
      "Approx 40% complete for SRR1822154_1.fastq.gz\n",
      "Approx 45% complete for SRR1822154_1.fastq.gz\n",
      "Approx 50% complete for SRR1822154_1.fastq.gz\n",
      "Approx 55% complete for SRR1822154_1.fastq.gz\n",
      "Approx 60% complete for SRR1822154_1.fastq.gz\n",
      "Approx 65% complete for SRR1822154_1.fastq.gz\n",
      "Approx 70% complete for SRR1822154_1.fastq.gz\n",
      "Approx 75% complete for SRR1822154_1.fastq.gz\n",
      "Approx 80% complete for SRR1822154_1.fastq.gz\n",
      "Approx 85% complete for SRR1822154_1.fastq.gz\n",
      "Approx 90% complete for SRR1822154_1.fastq.gz\n",
      "Approx 95% complete for SRR1822154_1.fastq.gz\n",
      "Approx 100% complete for SRR1822154_1.fastq.gz\n",
      "Analysis complete for SRR1822154_1.fastq.gz\n",
      "Started analysis of SRR1822157_1.fastq.gz\n",
      "Approx 5% complete for SRR1822157_1.fastq.gz\n",
      "Approx 10% complete for SRR1822157_1.fastq.gz\n",
      "Approx 15% complete for SRR1822157_1.fastq.gz\n",
      "Approx 20% complete for SRR1822157_1.fastq.gz\n",
      "Approx 25% complete for SRR1822157_1.fastq.gz\n",
      "Approx 30% complete for SRR1822157_1.fastq.gz\n",
      "Approx 35% complete for SRR1822157_1.fastq.gz\n",
      "Approx 40% complete for SRR1822157_1.fastq.gz\n",
      "Approx 45% complete for SRR1822157_1.fastq.gz\n",
      "Approx 50% complete for SRR1822157_1.fastq.gz\n",
      "Approx 55% complete for SRR1822157_1.fastq.gz\n",
      "Approx 60% complete for SRR1822157_1.fastq.gz\n",
      "Approx 65% complete for SRR1822157_1.fastq.gz\n",
      "Approx 70% complete for SRR1822157_1.fastq.gz\n",
      "Approx 75% complete for SRR1822157_1.fastq.gz\n",
      "Approx 80% complete for SRR1822157_1.fastq.gz\n",
      "Approx 85% complete for SRR1822157_1.fastq.gz\n",
      "Approx 90% complete for SRR1822157_1.fastq.gz\n",
      "Approx 95% complete for SRR1822157_1.fastq.gz\n",
      "Approx 100% complete for SRR1822157_1.fastq.gz\n",
      "Analysis complete for SRR1822157_1.fastq.gz\n",
      "Started analysis of SRR1822158_1.fastq.gz\n",
      "Approx 5% complete for SRR1822158_1.fastq.gz\n",
      "Approx 10% complete for SRR1822158_1.fastq.gz\n",
      "Approx 15% complete for SRR1822158_1.fastq.gz\n",
      "Approx 20% complete for SRR1822158_1.fastq.gz\n",
      "Approx 25% complete for SRR1822158_1.fastq.gz\n",
      "Approx 30% complete for SRR1822158_1.fastq.gz\n",
      "Approx 35% complete for SRR1822158_1.fastq.gz\n",
      "Approx 40% complete for SRR1822158_1.fastq.gz\n",
      "Approx 45% complete for SRR1822158_1.fastq.gz\n",
      "Approx 50% complete for SRR1822158_1.fastq.gz\n",
      "Approx 55% complete for SRR1822158_1.fastq.gz\n",
      "Approx 60% complete for SRR1822158_1.fastq.gz\n",
      "Approx 65% complete for SRR1822158_1.fastq.gz\n",
      "Approx 70% complete for SRR1822158_1.fastq.gz\n",
      "Approx 75% complete for SRR1822158_1.fastq.gz\n",
      "Approx 80% complete for SRR1822158_1.fastq.gz\n",
      "Approx 85% complete for SRR1822158_1.fastq.gz\n",
      "Approx 90% complete for SRR1822158_1.fastq.gz\n",
      "Approx 95% complete for SRR1822158_1.fastq.gz\n",
      "Approx 100% complete for SRR1822158_1.fastq.gz\n",
      "Analysis complete for SRR1822158_1.fastq.gz\n",
      "Started analysis of SRR1822153_2.fastq.gz\n",
      "Approx 5% complete for SRR1822153_2.fastq.gz\n",
      "Approx 10% complete for SRR1822153_2.fastq.gz\n",
      "Approx 15% complete for SRR1822153_2.fastq.gz\n",
      "Approx 20% complete for SRR1822153_2.fastq.gz\n",
      "Approx 25% complete for SRR1822153_2.fastq.gz\n",
      "Approx 30% complete for SRR1822153_2.fastq.gz\n",
      "Approx 35% complete for SRR1822153_2.fastq.gz\n",
      "Approx 40% complete for SRR1822153_2.fastq.gz\n",
      "Approx 45% complete for SRR1822153_2.fastq.gz\n",
      "Approx 50% complete for SRR1822153_2.fastq.gz\n",
      "Approx 55% complete for SRR1822153_2.fastq.gz\n",
      "Approx 60% complete for SRR1822153_2.fastq.gz\n",
      "Approx 65% complete for SRR1822153_2.fastq.gz\n",
      "Approx 70% complete for SRR1822153_2.fastq.gz\n",
      "Approx 75% complete for SRR1822153_2.fastq.gz\n",
      "Approx 80% complete for SRR1822153_2.fastq.gz\n",
      "Approx 85% complete for SRR1822153_2.fastq.gz\n",
      "Approx 90% complete for SRR1822153_2.fastq.gz\n",
      "Approx 95% complete for SRR1822153_2.fastq.gz\n",
      "Approx 100% complete for SRR1822153_2.fastq.gz\n",
      "Analysis complete for SRR1822153_2.fastq.gz\n",
      "Started analysis of SRR1822154_2.fastq.gz\n",
      "Approx 5% complete for SRR1822154_2.fastq.gz\n",
      "Approx 10% complete for SRR1822154_2.fastq.gz\n",
      "Approx 15% complete for SRR1822154_2.fastq.gz\n",
      "Approx 20% complete for SRR1822154_2.fastq.gz\n",
      "Approx 25% complete for SRR1822154_2.fastq.gz\n",
      "Approx 30% complete for SRR1822154_2.fastq.gz\n",
      "Approx 35% complete for SRR1822154_2.fastq.gz\n",
      "Approx 40% complete for SRR1822154_2.fastq.gz\n",
      "Approx 45% complete for SRR1822154_2.fastq.gz\n",
      "Approx 50% complete for SRR1822154_2.fastq.gz\n",
      "Approx 55% complete for SRR1822154_2.fastq.gz\n",
      "Approx 60% complete for SRR1822154_2.fastq.gz\n",
      "Approx 65% complete for SRR1822154_2.fastq.gz\n",
      "Approx 70% complete for SRR1822154_2.fastq.gz\n",
      "Approx 75% complete for SRR1822154_2.fastq.gz\n",
      "Approx 80% complete for SRR1822154_2.fastq.gz\n",
      "Approx 85% complete for SRR1822154_2.fastq.gz\n",
      "Approx 90% complete for SRR1822154_2.fastq.gz\n",
      "Approx 95% complete for SRR1822154_2.fastq.gz\n",
      "Approx 100% complete for SRR1822154_2.fastq.gz\n",
      "Analysis complete for SRR1822154_2.fastq.gz\n",
      "Started analysis of SRR1822157_2.fastq.gz\n",
      "Approx 5% complete for SRR1822157_2.fastq.gz\n",
      "Approx 10% complete for SRR1822157_2.fastq.gz\n",
      "Approx 15% complete for SRR1822157_2.fastq.gz\n",
      "Approx 20% complete for SRR1822157_2.fastq.gz\n",
      "Approx 25% complete for SRR1822157_2.fastq.gz\n",
      "Approx 30% complete for SRR1822157_2.fastq.gz\n",
      "Approx 35% complete for SRR1822157_2.fastq.gz\n",
      "Approx 40% complete for SRR1822157_2.fastq.gz\n",
      "Approx 45% complete for SRR1822157_2.fastq.gz\n",
      "Approx 50% complete for SRR1822157_2.fastq.gz\n",
      "Approx 55% complete for SRR1822157_2.fastq.gz\n",
      "Approx 60% complete for SRR1822157_2.fastq.gz\n",
      "Approx 65% complete for SRR1822157_2.fastq.gz\n",
      "Approx 70% complete for SRR1822157_2.fastq.gz\n",
      "Approx 75% complete for SRR1822157_2.fastq.gz\n",
      "Approx 80% complete for SRR1822157_2.fastq.gz\n",
      "Approx 85% complete for SRR1822157_2.fastq.gz\n",
      "Approx 90% complete for SRR1822157_2.fastq.gz\n",
      "Approx 95% complete for SRR1822157_2.fastq.gz\n",
      "Approx 100% complete for SRR1822157_2.fastq.gz\n",
      "Analysis complete for SRR1822157_2.fastq.gz\n",
      "Started analysis of SRR1822158_2.fastq.gz\n",
      "Approx 5% complete for SRR1822158_2.fastq.gz\n",
      "Approx 10% complete for SRR1822158_2.fastq.gz\n",
      "Approx 15% complete for SRR1822158_2.fastq.gz\n",
      "Approx 20% complete for SRR1822158_2.fastq.gz\n",
      "Approx 25% complete for SRR1822158_2.fastq.gz\n",
      "Approx 30% complete for SRR1822158_2.fastq.gz\n",
      "Approx 35% complete for SRR1822158_2.fastq.gz\n",
      "Approx 40% complete for SRR1822158_2.fastq.gz\n",
      "Approx 45% complete for SRR1822158_2.fastq.gz\n",
      "Approx 50% complete for SRR1822158_2.fastq.gz\n",
      "Approx 55% complete for SRR1822158_2.fastq.gz\n",
      "Approx 60% complete for SRR1822158_2.fastq.gz\n",
      "Approx 65% complete for SRR1822158_2.fastq.gz\n",
      "Approx 70% complete for SRR1822158_2.fastq.gz\n",
      "Approx 75% complete for SRR1822158_2.fastq.gz\n",
      "Approx 80% complete for SRR1822158_2.fastq.gz\n",
      "Approx 85% complete for SRR1822158_2.fastq.gz\n",
      "Approx 90% complete for SRR1822158_2.fastq.gz\n",
      "Approx 95% complete for SRR1822158_2.fastq.gz\n",
      "Approx 100% complete for SRR1822158_2.fastq.gz\n",
      "Analysis complete for SRR1822158_2.fastq.gz\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p output/stats/raw_fastQC/\n",
    "fastqc -o output/stats/raw_fastQC/ data/raw_reads/SRR182215{3,4,7,8}_1.fastq.gz data/raw_reads/SRR182215{3,4,7,8}_2.fastq.gz "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If you have multiple samples, multiQC is a good way to look at the global quality over your samples."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=751489;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.11\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/mag/ATACseq/TestJupyterNotebook/ATAC-seq/output/stats/raw_fastQC\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m16/16\u001b[0m   \n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 8 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : output/stats/raw_fastQC/multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : output/stats/raw_fastQC/multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "multiqc output/stats/raw_fastQC -o output/stats/raw_fastQC"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Trimming\n",
    "\n",
    "The fastq files are trimmed using Trimmomatic v0.39 to remove any adapter sequences in the reads caused by read through associated with DNA fragments shorter in size than the read length being sequenced.\n",
    "\n",
    "This particular step need to be adapted depending on your exact data and need. We found these options to be the most efficient in the case of ATAC-Seq. If you want to know more we encourage you to read the [documentation](http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf) of the Trimmomatic tool in order to find the best possible use for your case.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TrimmomaticPE: Started with arguments:2004l\n",
      " -threads 2 data/raw_reads/SRR1822153_1.fastq.gz data/raw_reads/SRR1822153_2.fastq.gz output/trimmed_files/SRR1822153_1_trimmed.fastq output/trimmed_files/SRR1822153_1_trimmed_unpaired.fa output/trimmed_files/SRR1822153_2_trimmed.fastq output/trimmed_files/SRR1822153_2_trimmed_unpaired.fa ILLUMINACLIP:data/adapters/NexteraPE-PE.fa:2:30:10:1:true TRAILING:3 SLIDINGWINDOW:4:15\n",
      "Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 100000 Both Surviving: 99529 (99,53%) Forward Only Surviving: 419 (0,42%) Reverse Only Surviving: 50 (0,05%) Dropped: 2 (0,00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_reads/SRR1822154_1.fastq.gz data/raw_reads/SRR1822154_2.fastq.gz output/trimmed_files/SRR1822154_1_trimmed.fastq output/trimmed_files/SRR1822154_1_trimmed_unpaired.fa output/trimmed_files/SRR1822154_2_trimmed.fastq output/trimmed_files/SRR1822154_2_trimmed_unpaired.fa ILLUMINACLIP:data/adapters/NexteraPE-PE.fa:2:30:10:1:true TRAILING:3 SLIDINGWINDOW:4:15\n",
      "Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 100000 Both Surviving: 99538 (99,54%) Forward Only Surviving: 415 (0,41%) Reverse Only Surviving: 47 (0,05%) Dropped: 0 (0,00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_reads/SRR1822157_1.fastq.gz data/raw_reads/SRR1822157_2.fastq.gz output/trimmed_files/SRR1822157_1_trimmed.fastq output/trimmed_files/SRR1822157_1_trimmed_unpaired.fa output/trimmed_files/SRR1822157_2_trimmed.fastq output/trimmed_files/SRR1822157_2_trimmed_unpaired.fa ILLUMINACLIP:data/adapters/NexteraPE-PE.fa:2:30:10:1:true TRAILING:3 SLIDINGWINDOW:4:15\n",
      "Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 100000 Both Surviving: 99576 (99,58%) Forward Only Surviving: 385 (0,39%) Reverse Only Surviving: 39 (0,04%) Dropped: 0 (0,00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "TrimmomaticPE: Started with arguments:\n",
      " -threads 2 data/raw_reads/SRR1822158_1.fastq.gz data/raw_reads/SRR1822158_2.fastq.gz output/trimmed_files/SRR1822158_1_trimmed.fastq output/trimmed_files/SRR1822158_1_trimmed_unpaired.fa output/trimmed_files/SRR1822158_2_trimmed.fastq output/trimmed_files/SRR1822158_2_trimmed_unpaired.fa ILLUMINACLIP:data/adapters/NexteraPE-PE.fa:2:30:10:1:true TRAILING:3 SLIDINGWINDOW:4:15\n",
      "Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'\n",
      "Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'\n",
      "ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences\n",
      "Quality encoding detected as phred33\n",
      "Input Read Pairs: 100000 Both Surviving: 99484 (99,48%) Forward Only Surviving: 466 (0,47%) Reverse Only Surviving: 48 (0,05%) Dropped: 2 (0,00%)\n",
      "TrimmomaticPE: Completed successfully\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p output/trimmed_files\n",
    "for i in {3,4,7,8}; \n",
    "do trimmomatic PE -threads 2 data/raw_reads/SRR182215${i}_1.fastq.gz data/raw_reads/SRR182215${i}_2.fastq.gz output/trimmed_files/SRR182215${i}_1_trimmed.fastq output/trimmed_files/SRR182215${i}_1_trimmed_unpaired.fa output/trimmed_files/SRR182215${i}_2_trimmed.fastq output/trimmed_files/SRR182215${i}_2_trimmed_unpaired.fa ILLUMINACLIP:data/adapters/NexteraPE-PE.fa:2:30:10:1:true TRAILING:3 SLIDINGWINDOW:4:15;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Alignement\n",
    "\n",
    "This step must also be adapted to your particular experiment, namely the genomes used may vary and the aligner used as well as the alignment options can be optimized.\n",
    "\n",
    "In our case, alignment is performed by bowtie2.4.4. \n",
    "\n",
    "### Bowtie2 Index\n",
    "\n",
    "In order to align reads to genome(s) and particular cellular compartments (i.e. mitochondria), it is necessary to build a bowtie index for each of them and potential confounders. In the end, only the reads mapping solely to the target of interest will be saved. In our case, the mitochondrial hits will be removed.\n",
    "\n",
    "Bowtie2 necessitate and index to be built before the alignment can be done. By default, the exact method used is chosen depending on the machine doing the computation, please read the [documentation of bowtie2-build](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml#the-bowtie2-build-indexer) for further information, and be careful when repeating a study on different machines.\n",
    "\n",
    "If you already have a bowtie index built for this type of alignment, this step can be skipped.\n",
    "\n",
    "In this example, we start from the fasta of a complete genome that we split into chromosomes using a small bash script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bash scripts/split_reference.sh reference/genome.fa"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Then we build one index for the mitochondrie, and one index for the other chromosomes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd reference\n",
    "bowtie2-build genome.fa_MT yeast_MT > MT_index.log\n",
    "bowtie2-build genome.fa_I,genome.fa_II,genome.fa_III,genome.fa_IV,genome.fa_V,genome.fa_VI,genome.fa_VII,genome.fa_VIII,genome.fa_IX,genome.fa_X,genome.fa_XI,genome.fa_XII,genome.fa_XIII,genome.fa_XIV,genome.fa_XV,genome.fa_XVI yeast_NC > NC_index.log\n",
    "cd .. # to go back to ATAC-seq folder. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Alignment\n",
    "\n",
    "#### Removal of Mitochondrial DNA\n",
    "\n",
    "Let's breakdown the various options used:\n",
    "\n",
    "* `-p 4` We are using 4 threads for alignment. This option can also be accessed via `--threads`.\n",
    "* `-X 2000` The maximum fragment length accepted between paired end. This number should be adapted depending on your sequencing protocol.\n",
    "* `--very-sensitive` Preset option corresponding to `-D 20 -R 3 -N 0 -L 20 -i S,1,0.50`.\n",
    "* `-x` Selected bowtie index.\n",
    "*  `-1 and -2` The input fastq files in the case of paired-end alignment\n",
    "*  `-S` File in which to write the output SAM file of the aligned reads. This file is not useful but printing it allows us to use less memory.\n",
    "*  `--un-conc` We only keep pairs that DIDN'T align on mitochondrial genome.\n",
    "\n",
    "for more informations and option, you can run `bowtie2 --help`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822153.917955 NS500418:8:H03J9AFXX:1:21311:1438:11285/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.917955 NS500418:8:H03J9AFXX:1:21311:1438:11285/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1868166 NS500418:8:H03J9AFXX:3:11403:26762:7214/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1868166 NS500418:8:H03J9AFXX:3:11403:26762:7214/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3438736 NS500418:8:H03J9AFXX:4:21405:14687:17026/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3438736 NS500418:8:H03J9AFXX:4:21405:14687:17026/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1816668 NS500418:8:H03J9AFXX:2:21311:14382:9820/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1816668 NS500418:8:H03J9AFXX:2:21311:14382:9820/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.676592 NS500418:8:H03J9AFXX:1:21204:18996:17988/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.676592 NS500418:8:H03J9AFXX:1:21204:18996:17988/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.467856 NS500418:8:H03J9AFXX:1:21101:23818:2679/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.467856 NS500418:8:H03J9AFXX:1:21101:23818:2679/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.605522 NS500418:8:H03J9AFXX:1:21111:22965:12905/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.605522 NS500418:8:H03J9AFXX:1:21111:22965:12905/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3663238 NS500418:8:H03J9AFXX:4:21509:1551:6318/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3663238 NS500418:8:H03J9AFXX:4:21509:1551:6318/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2569466 NS500418:8:H03J9AFXX:3:21505:25326:4177/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2569466 NS500418:8:H03J9AFXX:3:21505:25326:4177/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3529729 NS500418:8:H03J9AFXX:4:21411:24743:19767/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3529729 NS500418:8:H03J9AFXX:4:21411:24743:19767/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1250538 NS500418:8:H03J9AFXX:2:11301:3744:14775/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1250538 NS500418:8:H03J9AFXX:2:11301:3744:14775/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2272841 NS500418:8:H03J9AFXX:3:11608:2975:2109/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2272841 NS500418:8:H03J9AFXX:3:11608:2975:2109/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2660323 NS500418:8:H03J9AFXX:3:21511:23369:11084/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2660323 NS500418:8:H03J9AFXX:3:21511:23369:11084/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.533129 NS500418:8:H03J9AFXX:1:21106:4356:4200/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.533129 NS500418:8:H03J9AFXX:1:21106:4356:4200/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3708432 NS500418:8:H03J9AFXX:4:21512:23237:9254/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3708432 NS500418:8:H03J9AFXX:4:21512:23237:9254/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.223800 NS500418:8:H03J9AFXX:1:11206:22327:10008/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.223800 NS500418:8:H03J9AFXX:1:11206:22327:10008/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2538328 NS500418:8:H03J9AFXX:3:21503:10909:1368/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2538328 NS500418:8:H03J9AFXX:3:21503:10909:1368/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1806093 NS500418:8:H03J9AFXX:2:21310:9636:12403/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1806093 NS500418:8:H03J9AFXX:2:21310:9636:12403/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3196257 NS500418:8:H03J9AFXX:4:11512:18377:14844/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3196257 NS500418:8:H03J9AFXX:4:11512:18377:14844/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2854196 NS500418:8:H03J9AFXX:3:21612:14487:14227/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2854196 NS500418:8:H03J9AFXX:3:21612:14487:14227/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.201769 NS500418:8:H03J9AFXX:1:11204:9726:17120/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.201769 NS500418:8:H03J9AFXX:1:11204:9726:17120/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1735879 NS500418:8:H03J9AFXX:2:21304:1721:20258/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1735879 NS500418:8:H03J9AFXX:2:21304:1721:20258/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.909146 NS500418:8:H03J9AFXX:1:21310:14603:16663/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.909146 NS500418:8:H03J9AFXX:1:21310:14603:16663/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.839160 NS500418:8:H03J9AFXX:1:21305:17462:5295/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.839160 NS500418:8:H03J9AFXX:1:21305:17462:5295/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2645730 NS500418:8:H03J9AFXX:3:21510:26677:10521/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2645730 NS500418:8:H03J9AFXX:3:21510:26677:10521/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.37567 NS500418:8:H03J9AFXX:1:11104:8888:7453/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.37567 NS500418:8:H03J9AFXX:1:11104:8888:7453/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.505219 NS500418:8:H03J9AFXX:1:21104:14594:2028/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.505219 NS500418:8:H03J9AFXX:1:21104:14594:2028/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.295973 NS500418:8:H03J9AFXX:1:11211:14264:14974/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.295973 NS500418:8:H03J9AFXX:1:11211:14264:14974/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2822565 NS500418:8:H03J9AFXX:3:21610:17626:12598/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2822565 NS500418:8:H03J9AFXX:3:21610:17626:12598/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.847550 NS500418:8:H03J9AFXX:1:21305:9231:17934/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.847550 NS500418:8:H03J9AFXX:1:21305:9231:17934/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.537633 NS500418:8:H03J9AFXX:1:21106:8850:11130/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822153.537633 NS500418:8:H03J9AFXX:1:21106:8850:11130/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2054146 NS500418:8:H03J9AFXX:3:11504:20083:14575/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2054146 NS500418:8:H03J9AFXX:3:11504:20083:14575/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3088362 NS500418:8:H03J9AFXX:4:11505:18956:5107/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3088362 NS500418:8:H03J9AFXX:4:11505:18956:5107/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3641787 NS500418:8:H03J9AFXX:4:21507:25609:16184/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3641787 NS500418:8:H03J9AFXX:4:21507:25609:16184/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3522657 NS500418:8:H03J9AFXX:4:21411:5956:10509/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3522657 NS500418:8:H03J9AFXX:4:21411:5956:10509/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.176802 NS500418:8:H03J9AFXX:1:11203:17694:1083/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.176802 NS500418:8:H03J9AFXX:1:11203:17694:1083/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.474080 NS500418:8:H03J9AFXX:1:21101:21989:12890/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.474080 NS500418:8:H03J9AFXX:1:21101:21989:12890/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.1585486 NS500418:8:H03J9AFXX:2:21205:15721:6492/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.1585486 NS500418:8:H03J9AFXX:2:21205:15721:6492/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1086033 NS500418:8:H03J9AFXX:2:11201:25427:6788/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1086033 NS500418:8:H03J9AFXX:2:11201:25427:6788/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3720432 NS500418:8:H03J9AFXX:4:21601:24867:6709/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3720432 NS500418:8:H03J9AFXX:4:21601:24867:6709/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2131036 NS500418:8:H03J9AFXX:3:11509:23890:19447/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2131036 NS500418:8:H03J9AFXX:3:11509:23890:19447/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2830271 NS500418:8:H03J9AFXX:3:21611:11249:2527/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2830271 NS500418:8:H03J9AFXX:3:21611:11249:2527/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.536918 NS500418:8:H03J9AFXX:1:21106:26351:10081/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.536918 NS500418:8:H03J9AFXX:1:21106:26351:10081/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3832705 NS500418:8:H03J9AFXX:4:21609:23511:12958/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3832705 NS500418:8:H03J9AFXX:4:21609:23511:12958/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.67053 NS500418:8:H03J9AFXX:1:11106:26037:14266/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.67053 NS500418:8:H03J9AFXX:1:11106:26037:14266/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1221778 NS500418:8:H03J9AFXX:2:11211:17153:11176/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1221778 NS500418:8:H03J9AFXX:2:11211:17153:11176/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3633885 NS500418:8:H03J9AFXX:4:21507:25189:5420/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3633885 NS500418:8:H03J9AFXX:4:21507:25189:5420/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.316883 NS500418:8:H03J9AFXX:1:11301:14152:6465/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.316883 NS500418:8:H03J9AFXX:1:11301:14152:6465/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2504426 NS500418:8:H03J9AFXX:3:21412:3654:14250/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2504426 NS500418:8:H03J9AFXX:3:21412:3654:14250/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.107818 NS500418:8:H03J9AFXX:1:11109:2948:16944/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.107818 NS500418:8:H03J9AFXX:1:11109:2948:16944/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.981689 NS500418:8:H03J9AFXX:2:11104:3645:17951/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.981689 NS500418:8:H03J9AFXX:2:11104:3645:17951/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3623996 NS500418:8:H03J9AFXX:4:21506:21993:11541/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3623996 NS500418:8:H03J9AFXX:4:21506:21993:11541/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1026570 NS500418:8:H03J9AFXX:2:11108:24898:11265/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1026570 NS500418:8:H03J9AFXX:2:11108:24898:11265/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1983053 NS500418:8:H03J9AFXX:3:11411:1110:15611/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1983053 NS500418:8:H03J9AFXX:3:11411:1110:15611/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3665431 NS500418:8:H03J9AFXX:4:21509:20222:9473/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3665431 NS500418:8:H03J9AFXX:4:21509:20222:9473/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2135201 NS500418:8:H03J9AFXX:3:11510:25853:5664/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2135201 NS500418:8:H03J9AFXX:3:11510:25853:5664/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1557499 NS500418:8:H03J9AFXX:2:21203:17716:2968/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1557499 NS500418:8:H03J9AFXX:2:21203:17716:2968/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3679704 NS500418:8:H03J9AFXX:4:21510:13451:9179/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3679704 NS500418:8:H03J9AFXX:4:21510:13451:9179/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2067676 NS500418:8:H03J9AFXX:3:11505:26301:13158/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2067676 NS500418:8:H03J9AFXX:3:11505:26301:13158/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.2458807 NS500418:8:H03J9AFXX:3:21409:22609:3969/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.2458807 NS500418:8:H03J9AFXX:3:21409:22609:3969/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2553492 NS500418:8:H03J9AFXX:3:21504:25810:2359/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822153.2553492 NS500418:8:H03J9AFXX:3:21504:25810:2359/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3061818 NS500418:8:H03J9AFXX:4:11503:20950:8106/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3061818 NS500418:8:H03J9AFXX:4:11503:20950:8106/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3050149 NS500418:8:H03J9AFXX:4:11502:6845:11878/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3050149 NS500418:8:H03J9AFXX:4:11502:6845:11878/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.87379 NS500418:8:H03J9AFXX:1:11108:6366:6011/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.87379 NS500418:8:H03J9AFXX:1:11108:6366:6011/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1401081 NS500418:8:H03J9AFXX:2:21102:6147:5013/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1401081 NS500418:8:H03J9AFXX:2:21102:6147:5013/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3089135 NS500418:8:H03J9AFXX:4:11505:11736:6079/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3089135 NS500418:8:H03J9AFXX:4:11505:11736:6079/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1651556 NS500418:8:H03J9AFXX:2:21210:14158:8271/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1651556 NS500418:8:H03J9AFXX:2:21210:14158:8271/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3578127 NS500418:8:H03J9AFXX:4:21503:24969:6541/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3578127 NS500418:8:H03J9AFXX:4:21503:24969:6541/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.576531 NS500418:8:H03J9AFXX:1:21109:23867:9153/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.576531 NS500418:8:H03J9AFXX:1:21109:23867:9153/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.897382 NS500418:8:H03J9AFXX:1:21309:17004:17628/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.897382 NS500418:8:H03J9AFXX:1:21309:17004:17628/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2639655 NS500418:8:H03J9AFXX:3:21510:4530:1922/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2639655 NS500418:8:H03J9AFXX:3:21510:4530:1922/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3828546 NS500418:8:H03J9AFXX:4:21609:15182:6959/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3828546 NS500418:8:H03J9AFXX:4:21609:15182:6959/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2693670 NS500418:8:H03J9AFXX:3:21601:24058:18516/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2693670 NS500418:8:H03J9AFXX:3:21601:24058:18516/2' because it was < 2 characters long\n",
      "99529 reads; of these:\n",
      "  99529 (100.00%) were paired; of these:\n",
      "    98451 (98.92%) aligned concordantly 0 times\n",
      "    851 (0.86%) aligned concordantly exactly 1 time\n",
      "    227 (0.23%) aligned concordantly >1 times\n",
      "    ----\n",
      "    98451 pairs aligned concordantly 0 times; of these:\n",
      "      0 (0.00%) aligned discordantly 1 time\n",
      "    ----\n",
      "    98451 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      196902 mates make up the pairs; of these:\n",
      "        195868 (99.47%) aligned 0 times\n",
      "        78 (0.04%) aligned exactly 1 time\n",
      "        956 (0.49%) aligned >1 times\n",
      "1.60% overall alignment rate\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6619810 NS500418:8:H03J9AFXX:3:21512:15180:17756/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6619810 NS500418:8:H03J9AFXX:3:21512:15180:17756/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5375876 NS500418:8:H03J9AFXX:3:11601:26208:12883/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5375876 NS500418:8:H03J9AFXX:3:11601:26208:12883/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4301366 NS500418:8:H03J9AFXX:2:21306:18235:9660/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4301366 NS500418:8:H03J9AFXX:2:21306:18235:9660/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012279 NS500418:8:H03J9AFXX:4:21506:4266:19438/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012279 NS500418:8:H03J9AFXX:4:21506:4266:19438/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7297360 NS500418:8:H03J9AFXX:4:11407:25658:3755/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7297360 NS500418:8:H03J9AFXX:4:11407:25658:3755/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6797166 NS500418:8:H03J9AFXX:3:21605:12156:17787/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6797166 NS500418:8:H03J9AFXX:3:21605:12156:17787/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.912465 NS500418:8:H03J9AFXX:1:11305:11839:13985/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.912465 NS500418:8:H03J9AFXX:1:11305:11839:13985/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4743093 NS500418:8:H03J9AFXX:3:11407:5789:15822/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4743093 NS500418:8:H03J9AFXX:3:11407:5789:15822/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7305707 NS500418:8:H03J9AFXX:4:11407:1799:8175/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7305707 NS500418:8:H03J9AFXX:4:11407:1799:8175/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4874276 NS500418:8:H03J9AFXX:3:11411:24601:13205/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4874276 NS500418:8:H03J9AFXX:3:11411:24601:13205/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6069398 NS500418:8:H03J9AFXX:3:21409:17500:2979/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6069398 NS500418:8:H03J9AFXX:3:21409:17500:2979/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9506063 NS500418:8:H03J9AFXX:4:21609:18368:5721/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9506063 NS500418:8:H03J9AFXX:4:21609:18368:5721/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.7879513 NS500418:8:H03J9AFXX:4:11511:25797:10154/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.7879513 NS500418:8:H03J9AFXX:4:11511:25797:10154/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012808 NS500418:8:H03J9AFXX:4:21506:18761:19698/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012808 NS500418:8:H03J9AFXX:4:21506:18761:19698/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3664145 NS500418:8:H03J9AFXX:2:21110:20419:1624/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3664145 NS500418:8:H03J9AFXX:2:21110:20419:1624/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.3513824 NS500418:8:H03J9AFXX:2:21104:2431:19125/1' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #1 of read 'SRR1822154.3513824 NS500418:8:H03J9AFXX:2:21104:2431:19125/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7904923 NS500418:8:H03J9AFXX:4:11512:23465:4400/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7904923 NS500418:8:H03J9AFXX:4:11512:23465:4400/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5537589 NS500418:8:H03J9AFXX:3:11606:20305:6399/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5537589 NS500418:8:H03J9AFXX:3:11606:20305:6399/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.379073 NS500418:8:H03J9AFXX:1:11201:23092:9050/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.379073 NS500418:8:H03J9AFXX:1:11201:23092:9050/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4311319 NS500418:8:H03J9AFXX:2:21306:2974:15830/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4311319 NS500418:8:H03J9AFXX:2:21306:2974:15830/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2228424 NS500418:8:H03J9AFXX:1:21310:22521:19044/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2228424 NS500418:8:H03J9AFXX:1:21310:22521:19044/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.26644 NS500418:8:H03J9AFXX:1:11101:26863:20098/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.26644 NS500418:8:H03J9AFXX:1:11101:26863:20098/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2636123 NS500418:8:H03J9AFXX:2:11112:16593:10529/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2636123 NS500418:8:H03J9AFXX:2:11112:16593:10529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5940626 NS500418:8:H03J9AFXX:3:21405:19168:7424/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5940626 NS500418:8:H03J9AFXX:3:21405:19168:7424/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6726585 NS500418:8:H03J9AFXX:3:21603:23383:19601/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6726585 NS500418:8:H03J9AFXX:3:21603:23383:19601/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8514061 NS500418:8:H03J9AFXX:4:21405:7936:7343/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8514061 NS500418:8:H03J9AFXX:4:21405:7936:7343/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7869584 NS500418:8:H03J9AFXX:4:11511:21477:4857/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7869584 NS500418:8:H03J9AFXX:4:11511:21477:4857/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.431690 NS500418:8:H03J9AFXX:1:11202:3805:20065/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.431690 NS500418:8:H03J9AFXX:1:11202:3805:20065/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.9409189 NS500418:8:H03J9AFXX:4:21606:26701:11337/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.9409189 NS500418:8:H03J9AFXX:4:21606:26701:11337/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1532495 NS500418:8:H03J9AFXX:1:21201:22051:5369/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1532495 NS500418:8:H03J9AFXX:1:21201:22051:5369/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6795534 NS500418:8:H03J9AFXX:3:21605:25638:16990/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6795534 NS500418:8:H03J9AFXX:3:21605:25638:16990/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3162043 NS500418:8:H03J9AFXX:2:11304:18127:19777/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3162043 NS500418:8:H03J9AFXX:2:11304:18127:19777/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8098985 NS500418:8:H03J9AFXX:4:11605:13396:19467/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8098985 NS500418:8:H03J9AFXX:4:11605:13396:19467/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5269966 NS500418:8:H03J9AFXX:3:11510:13634:11671/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5269966 NS500418:8:H03J9AFXX:3:11510:13634:11671/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2516065 NS500418:8:H03J9AFXX:2:11108:24219:12077/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2516065 NS500418:8:H03J9AFXX:2:11108:24219:12077/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3324702 NS500418:8:H03J9AFXX:2:11310:3535:6190/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3324702 NS500418:8:H03J9AFXX:2:11310:3535:6190/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8266923 NS500418:8:H03J9AFXX:4:11610:15558:13142/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8266923 NS500418:8:H03J9AFXX:4:11610:15558:13142/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8572612 NS500418:8:H03J9AFXX:4:21406:18591:18319/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8572612 NS500418:8:H03J9AFXX:4:21406:18591:18319/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5247786 NS500418:8:H03J9AFXX:3:11509:16612:19219/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5247786 NS500418:8:H03J9AFXX:3:11509:16612:19219/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.246332 NS500418:8:H03J9AFXX:1:11109:4901:5414/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.246332 NS500418:8:H03J9AFXX:1:11109:4901:5414/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3520675 NS500418:8:H03J9AFXX:2:21105:2345:4258/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3520675 NS500418:8:H03J9AFXX:2:21105:2345:4258/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3391032 NS500418:8:H03J9AFXX:2:11312:7456:11981/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3391032 NS500418:8:H03J9AFXX:2:11312:7456:11981/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3039879 NS500418:8:H03J9AFXX:2:11212:16967:20392/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3039879 NS500418:8:H03J9AFXX:2:11212:16967:20392/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2357923 NS500418:8:H03J9AFXX:2:11103:18857:6625/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2357923 NS500418:8:H03J9AFXX:2:11103:18857:6625/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5878374 NS500418:8:H03J9AFXX:3:21403:7081:11334/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5878374 NS500418:8:H03J9AFXX:3:21403:7081:11334/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1266814 NS500418:8:H03J9AFXX:1:21104:24432:19972/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822154.1266814 NS500418:8:H03J9AFXX:1:21104:24432:19972/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6106920 NS500418:8:H03J9AFXX:3:21410:10571:5689/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6106920 NS500418:8:H03J9AFXX:3:21410:10571:5689/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4187359 NS500418:8:H03J9AFXX:2:21302:23042:15949/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4187359 NS500418:8:H03J9AFXX:2:21302:23042:15949/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5608565 NS500418:8:H03J9AFXX:3:11608:22684:3902/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5608565 NS500418:8:H03J9AFXX:3:11608:22684:3902/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2317140 NS500418:8:H03J9AFXX:2:11101:9968:18331/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2317140 NS500418:8:H03J9AFXX:2:11101:9968:18331/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2346183 NS500418:8:H03J9AFXX:2:11102:22622:18160/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2346183 NS500418:8:H03J9AFXX:2:11102:22622:18160/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.6820003 NS500418:8:H03J9AFXX:3:21606:14057:10145/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.6820003 NS500418:8:H03J9AFXX:3:21606:14057:10145/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6690186 NS500418:8:H03J9AFXX:3:21602:17108:18974/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6690186 NS500418:8:H03J9AFXX:3:21602:17108:18974/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8203240 NS500418:8:H03J9AFXX:4:11608:24474:17936/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8203240 NS500418:8:H03J9AFXX:4:11608:24474:17936/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4066696 NS500418:8:H03J9AFXX:2:21210:23966:19204/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4066696 NS500418:8:H03J9AFXX:2:21210:23966:19204/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4381296 NS500418:8:H03J9AFXX:2:21309:24821:1360/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4381296 NS500418:8:H03J9AFXX:2:21309:24821:1360/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.5491602 NS500418:8:H03J9AFXX:3:11605:5925:1859/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.5491602 NS500418:8:H03J9AFXX:3:11605:5925:1859/1' because it was < 2 characters long\n",
      "99538 reads; of these:\n",
      "  99538 (100.00%) were paired; of these:\n",
      "    98497 (98.95%) aligned concordantly 0 times\n",
      "    855 (0.86%) aligned concordantly exactly 1 time\n",
      "    186 (0.19%) aligned concordantly >1 times\n",
      "    ----\n",
      "    98497 pairs aligned concordantly 0 times; of these:\n",
      "      1 (0.00%) aligned discordantly 1 time\n",
      "    ----\n",
      "    98496 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      196992 mates make up the pairs; of these:\n",
      "        195953 (99.47%) aligned 0 times\n",
      "        83 (0.04%) aligned exactly 1 time\n",
      "        956 (0.49%) aligned >1 times\n",
      "1.57% overall alignment rate\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4244520 NS500418:8:H03J9AFXX:4:21404:13861:10491/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4244520 NS500418:8:H03J9AFXX:4:21404:13861:10491/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874583 NS500418:8:H03J9AFXX:1:21207:11553:10798/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874583 NS500418:8:H03J9AFXX:1:21207:11553:10798/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2403542 NS500418:8:H03J9AFXX:3:11409:22143:3243/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2403542 NS500418:8:H03J9AFXX:3:11409:22143:3243/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.194140 NS500418:8:H03J9AFXX:1:11201:21745:11469/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1235169 NS500418:8:H03J9AFXX:2:11106:25548:14740/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.194140 NS500418:8:H03J9AFXX:1:11201:21745:11469/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1235169 NS500418:8:H03J9AFXX:2:11106:25548:14740/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874842 NS500418:8:H03J9AFXX:1:21207:6471:11082/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874842 NS500418:8:H03J9AFXX:1:21207:6471:11082/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4466261 NS500418:8:H03J9AFXX:4:21504:14323:15574/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4466261 NS500418:8:H03J9AFXX:4:21504:14323:15574/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822157.1043285 NS500418:8:H03J9AFXX:1:21306:22089:1077/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822157.1043285 NS500418:8:H03J9AFXX:1:21306:22089:1077/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3558482 NS500418:8:H03J9AFXX:4:11401:7740:17153/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3558482 NS500418:8:H03J9AFXX:4:11401:7740:17153/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4291071 NS500418:8:H03J9AFXX:4:21406:24053:20054/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4291071 NS500418:8:H03J9AFXX:4:21406:24053:20054/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3155580 NS500418:8:H03J9AFXX:3:21503:23507:13435/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3155580 NS500418:8:H03J9AFXX:3:21503:23507:13435/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1952138 NS500418:8:H03J9AFXX:2:21205:2736:4863/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1952138 NS500418:8:H03J9AFXX:2:21205:2736:4863/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4566635 NS500418:8:H03J9AFXX:4:21510:7902:9560/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4566635 NS500418:8:H03J9AFXX:4:21510:7902:9560/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2596222 NS500418:8:H03J9AFXX:3:11507:19465:18428/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2596222 NS500418:8:H03J9AFXX:3:11507:19465:18428/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2469249 NS500418:8:H03J9AFXX:3:11501:4623:2573/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2469249 NS500418:8:H03J9AFXX:3:11501:4623:2573/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1947513 NS500418:8:H03J9AFXX:2:21204:21559:18573/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1947513 NS500418:8:H03J9AFXX:2:21204:21559:18573/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4778970 NS500418:8:H03J9AFXX:4:21610:4469:17529/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822157.4778970 NS500418:8:H03J9AFXX:4:21610:4469:17529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3517948 NS500418:8:H03J9AFXX:3:21611:25487:14102/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3517948 NS500418:8:H03J9AFXX:3:21611:25487:14102/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2940338 NS500418:8:H03J9AFXX:3:21402:1485:19745/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2940338 NS500418:8:H03J9AFXX:3:21402:1485:19745/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4022474 NS500418:8:H03J9AFXX:4:11604:10254:3180/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4022474 NS500418:8:H03J9AFXX:4:11604:10254:3180/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1964597 NS500418:8:H03J9AFXX:2:21205:5166:19904/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1964597 NS500418:8:H03J9AFXX:2:21205:5166:19904/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.246175 NS500418:8:H03J9AFXX:1:11204:17318:13066/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.246175 NS500418:8:H03J9AFXX:1:11204:17318:13066/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3571863 NS500418:8:H03J9AFXX:4:11402:15222:11807/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3571863 NS500418:8:H03J9AFXX:4:11402:15222:11807/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4707532 NS500418:8:H03J9AFXX:4:21606:2744:17315/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4707532 NS500418:8:H03J9AFXX:4:21606:2744:17315/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3589098 NS500418:8:H03J9AFXX:4:11403:21355:10333/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3589098 NS500418:8:H03J9AFXX:4:11403:21355:10333/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.579916 NS500418:8:H03J9AFXX:1:21101:1692:4498/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.579916 NS500418:8:H03J9AFXX:1:21101:1692:4498/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2643943 NS500418:8:H03J9AFXX:3:11510:5522:11322/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2643943 NS500418:8:H03J9AFXX:3:11510:5522:11322/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.311609 NS500418:8:H03J9AFXX:1:11208:7519:11193/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.311609 NS500418:8:H03J9AFXX:1:11208:7519:11193/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3568179 NS500418:8:H03J9AFXX:4:11402:9141:7864/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3568179 NS500418:8:H03J9AFXX:4:11402:9141:7864/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790589 NS500418:8:H03J9AFXX:4:11502:23067:16911/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790589 NS500418:8:H03J9AFXX:4:11502:23067:16911/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1527361 NS500418:8:H03J9AFXX:2:11212:18527:20335/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1527361 NS500418:8:H03J9AFXX:2:11212:18527:20335/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4305628 NS500418:8:H03J9AFXX:4:21407:26618:16283/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4305628 NS500418:8:H03J9AFXX:4:21407:26618:16283/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4564163 NS500418:8:H03J9AFXX:4:21510:14511:6815/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4564163 NS500418:8:H03J9AFXX:4:21510:14511:6815/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1424342 NS500418:8:H03J9AFXX:2:11206:21685:13980/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1424342 NS500418:8:H03J9AFXX:2:11206:21685:13980/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.602210 NS500418:8:H03J9AFXX:1:21102:20234:13842/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.602210 NS500418:8:H03J9AFXX:1:21102:20234:13842/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4272542 NS500418:8:H03J9AFXX:4:21405:5930:20354/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4272542 NS500418:8:H03J9AFXX:4:21405:5930:20354/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2869676 NS500418:8:H03J9AFXX:3:11611:7632:1955/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2869676 NS500418:8:H03J9AFXX:3:11611:7632:1955/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2896384 NS500418:8:H03J9AFXX:3:11612:20992:9899/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2896384 NS500418:8:H03J9AFXX:3:11612:20992:9899/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1606448 NS500418:8:H03J9AFXX:2:11306:3168:2556/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1606448 NS500418:8:H03J9AFXX:2:11306:3168:2556/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4449449 NS500418:8:H03J9AFXX:4:21503:1093:15969/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4449449 NS500418:8:H03J9AFXX:4:21503:1093:15969/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2650855 NS500418:8:H03J9AFXX:3:11510:12270:18842/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2650855 NS500418:8:H03J9AFXX:3:11510:12270:18842/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2012581 NS500418:8:H03J9AFXX:2:21208:12294:20282/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2012581 NS500418:8:H03J9AFXX:2:21208:12294:20282/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1135462 NS500418:8:H03J9AFXX:1:21311:15773:20238/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1135462 NS500418:8:H03J9AFXX:1:21311:15773:20238/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.182673 NS500418:8:H03J9AFXX:1:11112:3439:17433/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.182673 NS500418:8:H03J9AFXX:1:11112:3439:17433/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1101534 NS500418:8:H03J9AFXX:1:21309:26518:15716/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1101534 NS500418:8:H03J9AFXX:1:21309:26518:15716/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1291566 NS500418:8:H03J9AFXX:2:11110:19382:8350/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1291566 NS500418:8:H03J9AFXX:2:11110:19382:8350/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3436796 NS500418:8:H03J9AFXX:3:21607:4534:8800/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822157.3436796 NS500418:8:H03J9AFXX:3:21607:4534:8800/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2533842 NS500418:8:H03J9AFXX:3:11504:13926:11277/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2533842 NS500418:8:H03J9AFXX:3:11504:13926:11277/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3260827 NS500418:8:H03J9AFXX:3:21509:23447:11138/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3260827 NS500418:8:H03J9AFXX:3:21509:23447:11138/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790923 NS500418:8:H03J9AFXX:4:11502:2233:17239/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790923 NS500418:8:H03J9AFXX:4:11502:2233:17239/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2683443 NS500418:8:H03J9AFXX:3:11512:23587:15558/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2683443 NS500418:8:H03J9AFXX:3:11512:23587:15558/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3927746 NS500418:8:H03J9AFXX:4:11510:4583:10311/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3927746 NS500418:8:H03J9AFXX:4:11510:4583:10311/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.733070 NS500418:8:H03J9AFXX:1:21110:1441:18600/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.733070 NS500418:8:H03J9AFXX:1:21110:1441:18600/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4615013 NS500418:8:H03J9AFXX:4:21601:22050:4559/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4570459 NS500418:8:H03J9AFXX:4:21510:7398:13841/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4615013 NS500418:8:H03J9AFXX:4:21601:22050:4559/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4570459 NS500418:8:H03J9AFXX:4:21510:7398:13841/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.905288 NS500418:8:H03J9AFXX:1:21209:3978:8004/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.905288 NS500418:8:H03J9AFXX:1:21209:3978:8004/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4149075 NS500418:8:H03J9AFXX:4:11611:4775:5530/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4149075 NS500418:8:H03J9AFXX:4:11611:4775:5530/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.893037 NS500418:8:H03J9AFXX:1:21208:18960:13181/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.893037 NS500418:8:H03J9AFXX:1:21208:18960:13181/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.763997 NS500418:8:H03J9AFXX:1:21112:9756:17010/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.763997 NS500418:8:H03J9AFXX:1:21112:9756:17010/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.139747 NS500418:8:H03J9AFXX:1:11110:24875:4331/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.139747 NS500418:8:H03J9AFXX:1:11110:24875:4331/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790514 NS500418:8:H03J9AFXX:4:11502:10204:16838/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790514 NS500418:8:H03J9AFXX:4:11502:10204:16838/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1432839 NS500418:8:H03J9AFXX:2:11207:5542:4960/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1432839 NS500418:8:H03J9AFXX:2:11207:5542:4960/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.663627 NS500418:8:H03J9AFXX:1:21106:22227:12106/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.663627 NS500418:8:H03J9AFXX:1:21106:22227:12106/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1737749 NS500418:8:H03J9AFXX:2:21102:5051:19318/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1737749 NS500418:8:H03J9AFXX:2:21102:5051:19318/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.922149 NS500418:8:H03J9AFXX:1:21210:6982:8403/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.922149 NS500418:8:H03J9AFXX:1:21210:6982:8403/2' because it was < 2 characters long\n",
      "99576 reads; of these:\n",
      "  99576 (100.00%) were paired; of these:\n",
      "    99252 (99.67%) aligned concordantly 0 times\n",
      "    219 (0.22%) aligned concordantly exactly 1 time\n",
      "    105 (0.11%) aligned concordantly >1 times\n",
      "    ----\n",
      "    99252 pairs aligned concordantly 0 times; of these:\n",
      "      1 (0.00%) aligned discordantly 1 time\n",
      "    ----\n",
      "    99251 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      198502 mates make up the pairs; of these:\n",
      "        197627 (99.56%) aligned 0 times\n",
      "        65 (0.03%) aligned exactly 1 time\n",
      "        810 (0.41%) aligned >1 times\n",
      "0.77% overall alignment rate\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1515577 NS500418:8:H03J9AFXX:2:11109:2946:11317/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1515577 NS500418:8:H03J9AFXX:2:11109:2946:11317/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2841246 NS500418:8:H03J9AFXX:3:11408:13800:14719/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2841246 NS500418:8:H03J9AFXX:3:11408:13800:14719/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5213860 NS500418:8:H03J9AFXX:4:21412:25584:19892/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5213860 NS500418:8:H03J9AFXX:4:21412:25584:19892/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1574134 NS500418:8:H03J9AFXX:2:11112:25956:15203/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1574134 NS500418:8:H03J9AFXX:2:11112:25956:15203/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2722199 NS500418:8:H03J9AFXX:3:11402:24229:18703/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2722199 NS500418:8:H03J9AFXX:3:11402:24229:18703/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4729890 NS500418:8:H03J9AFXX:4:11602:19415:7772/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4729890 NS500418:8:H03J9AFXX:4:11602:19415:7772/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2223045 NS500418:8:H03J9AFXX:2:21112:9293:5545/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2223045 NS500418:8:H03J9AFXX:2:21112:9293:5545/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3992264 NS500418:8:H03J9AFXX:3:21603:23252:19490/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3992264 NS500418:8:H03J9AFXX:3:21603:23252:19490/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5317270 NS500418:8:H03J9AFXX:4:21505:25390:19471/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5317270 NS500418:8:H03J9AFXX:4:21505:25390:19471/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3441264 NS500418:8:H03J9AFXX:3:11612:25272:19343/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.3441264 NS500418:8:H03J9AFXX:3:11612:25272:19343/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4252580 NS500418:8:H03J9AFXX:4:11403:8072:14570/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4252580 NS500418:8:H03J9AFXX:4:11403:8072:14570/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5316087 NS500418:8:H03J9AFXX:4:21505:10289:18442/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5316087 NS500418:8:H03J9AFXX:4:21505:10289:18442/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3416966 NS500418:8:H03J9AFXX:3:11611:20418:17695/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3416966 NS500418:8:H03J9AFXX:3:11611:20418:17695/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1336914 NS500418:8:H03J9AFXX:1:21311:2806:10501/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1336914 NS500418:8:H03J9AFXX:1:21311:2806:10501/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2626174 NS500418:8:H03J9AFXX:2:21309:20094:19895/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2626174 NS500418:8:H03J9AFXX:2:21309:20094:19895/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5336362 NS500418:8:H03J9AFXX:4:21506:14405:18104/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5336362 NS500418:8:H03J9AFXX:4:21506:14405:18104/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4668943 NS500418:8:H03J9AFXX:4:11511:1694:7553/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4668943 NS500418:8:H03J9AFXX:4:11511:1694:7553/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4371848 NS500418:8:H03J9AFXX:4:11409:1882:4065/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4371848 NS500418:8:H03J9AFXX:4:11409:1882:4065/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5099006 NS500418:8:H03J9AFXX:4:21407:1428:15726/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5099006 NS500418:8:H03J9AFXX:4:21407:1428:15726/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.707160 NS500418:8:H03J9AFXX:1:21102:6801:6912/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.707160 NS500418:8:H03J9AFXX:1:21102:6801:6912/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4187963 NS500418:8:H03J9AFXX:3:21612:9346:16158/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4187963 NS500418:8:H03J9AFXX:3:21612:9346:16158/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4121224 NS500418:8:H03J9AFXX:3:21609:4528:15928/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4121224 NS500418:8:H03J9AFXX:3:21609:4528:15928/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4495875 NS500418:8:H03J9AFXX:4:11503:13529:4865/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4495875 NS500418:8:H03J9AFXX:4:11503:13529:4865/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2235112 NS500418:8:H03J9AFXX:2:21112:15560:18732/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2235112 NS500418:8:H03J9AFXX:2:21112:15560:18732/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142948 NS500418:8:H03J9AFXX:1:11108:23289:20264/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142948 NS500418:8:H03J9AFXX:1:11108:23289:20264/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.124102 NS500418:8:H03J9AFXX:1:11108:21213:1397/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.124102 NS500418:8:H03J9AFXX:1:11108:21213:1397/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2300328 NS500418:8:H03J9AFXX:2:21204:15766:9026/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2300328 NS500418:8:H03J9AFXX:2:21204:15766:9026/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1825373 NS500418:8:H03J9AFXX:2:11301:25074:14403/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1825373 NS500418:8:H03J9AFXX:2:11301:25074:14403/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4683494 NS500418:8:H03J9AFXX:4:11512:4340:1487/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4683494 NS500418:8:H03J9AFXX:4:11512:4340:1487/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1651720 NS500418:8:H03J9AFXX:2:11204:3330:15570/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1651720 NS500418:8:H03J9AFXX:2:11204:3330:15570/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3875592 NS500418:8:H03J9AFXX:3:21510:11640:5180/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3875592 NS500418:8:H03J9AFXX:3:21510:11640:5180/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2540914 NS500418:8:H03J9AFXX:2:21305:20539:7658/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2540914 NS500418:8:H03J9AFXX:2:21305:20539:7658/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4779982 NS500418:8:H03J9AFXX:4:11604:5735:18480/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4779982 NS500418:8:H03J9AFXX:4:11604:5735:18480/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5674531 NS500418:8:H03J9AFXX:4:21611:25817:10879/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5674531 NS500418:8:H03J9AFXX:4:21611:25817:10879/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3829550 NS500418:8:H03J9AFXX:3:21507:16566:20315/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3829550 NS500418:8:H03J9AFXX:3:21507:16566:20315/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3455197 NS500418:8:H03J9AFXX:3:21401:14672:13422/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3455197 NS500418:8:H03J9AFXX:3:21401:14672:13422/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4199278 NS500418:8:H03J9AFXX:4:11401:10874:6406/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4199278 NS500418:8:H03J9AFXX:4:11401:10874:6406/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2425736 NS500418:8:H03J9AFXX:2:21211:7354:2340/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2425736 NS500418:8:H03J9AFXX:2:21211:7354:2340/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5030491 NS500418:8:H03J9AFXX:4:21404:21612:13859/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5030491 NS500418:8:H03J9AFXX:4:21404:21612:13859/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2586653 NS500418:8:H03J9AFXX:2:21307:24676:17042/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.2586653 NS500418:8:H03J9AFXX:2:21307:24676:17042/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5295901 NS500418:8:H03J9AFXX:4:21504:5540:19050/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5295901 NS500418:8:H03J9AFXX:4:21504:5540:19050/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4674937 NS500418:8:H03J9AFXX:4:11511:2046:13054/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4674937 NS500418:8:H03J9AFXX:4:11511:2046:13054/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3885769 NS500418:8:H03J9AFXX:3:21510:10358:14634/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3885769 NS500418:8:H03J9AFXX:3:21510:10358:14634/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2834588 NS500418:8:H03J9AFXX:3:11408:6511:8304/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2834588 NS500418:8:H03J9AFXX:3:11408:6511:8304/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5661358 NS500418:8:H03J9AFXX:4:21610:22171:18112/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5661358 NS500418:8:H03J9AFXX:4:21610:22171:18112/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4943824 NS500418:8:H03J9AFXX:4:11612:5472:14299/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4943824 NS500418:8:H03J9AFXX:4:11612:5472:14299/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4177262 NS500418:8:H03J9AFXX:3:21612:15044:6655/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4177262 NS500418:8:H03J9AFXX:3:21612:15044:6655/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2387990 NS500418:8:H03J9AFXX:2:21209:6043:1941/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2387990 NS500418:8:H03J9AFXX:2:21209:6043:1941/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1171651 NS500418:8:H03J9AFXX:1:21302:25629:9670/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1171651 NS500418:8:H03J9AFXX:1:21302:25629:9670/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4010360 NS500418:8:H03J9AFXX:3:21604:24772:16656/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4010360 NS500418:8:H03J9AFXX:3:21604:24772:16656/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4538030 NS500418:8:H03J9AFXX:4:11505:10225:4596/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4538030 NS500418:8:H03J9AFXX:4:11505:10225:4596/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4974103 NS500418:8:H03J9AFXX:4:21402:15190:2856/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4974103 NS500418:8:H03J9AFXX:4:21402:15190:2856/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2637703 NS500418:8:H03J9AFXX:2:21310:9675:12704/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2637703 NS500418:8:H03J9AFXX:2:21310:9675:12704/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.126781 NS500418:8:H03J9AFXX:1:11108:23709:4021/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.126781 NS500418:8:H03J9AFXX:1:11108:23709:4021/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1775038 NS500418:8:H03J9AFXX:2:11211:6688:2776/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1775038 NS500418:8:H03J9AFXX:2:11211:6688:2776/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5392497 NS500418:8:H03J9AFXX:4:21509:9769:12463/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5392497 NS500418:8:H03J9AFXX:4:21509:9769:12463/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2412801 NS500418:8:H03J9AFXX:2:21210:13015:8588/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2412801 NS500418:8:H03J9AFXX:2:21210:13015:8588/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1829945 NS500418:8:H03J9AFXX:2:11301:12293:19483/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1829945 NS500418:8:H03J9AFXX:2:11301:12293:19483/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5546313 NS500418:8:H03J9AFXX:4:21605:2268:7223/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5546313 NS500418:8:H03J9AFXX:4:21605:2268:7223/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3099064 NS500418:8:H03J9AFXX:3:11508:1992:19314/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3099064 NS500418:8:H03J9AFXX:3:11508:1992:19314/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.209555 NS500418:8:H03J9AFXX:1:11112:6932:10529/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.209555 NS500418:8:H03J9AFXX:1:11112:6932:10529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1617141 NS500418:8:H03J9AFXX:2:11202:9871:19780/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1617141 NS500418:8:H03J9AFXX:2:11202:9871:19780/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.242207 NS500418:8:H03J9AFXX:1:11202:13972:4073/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.242207 NS500418:8:H03J9AFXX:1:11202:13972:4073/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.229514 NS500418:8:H03J9AFXX:1:11201:2447:11165/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.229514 NS500418:8:H03J9AFXX:1:11201:2447:11165/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3682078 NS500418:8:H03J9AFXX:3:21501:10891:2659/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3682078 NS500418:8:H03J9AFXX:3:21501:10891:2659/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3694129 NS500418:8:H03J9AFXX:3:21501:13111:14038/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3694129 NS500418:8:H03J9AFXX:3:21501:13111:14038/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2260158 NS500418:8:H03J9AFXX:2:21202:1137:6114/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2260158 NS500418:8:H03J9AFXX:2:21202:1137:6114/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142644 NS500418:8:H03J9AFXX:1:11108:26702:19961/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142644 NS500418:8:H03J9AFXX:1:11108:26702:19961/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3227533 NS500418:8:H03J9AFXX:3:11603:4656:6322/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3227533 NS500418:8:H03J9AFXX:3:11603:4656:6322/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4577305 NS500418:8:H03J9AFXX:4:11507:24943:2184/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4577305 NS500418:8:H03J9AFXX:4:11507:24943:2184/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.3525986 NS500418:8:H03J9AFXX:3:21405:9193:3657/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3525986 NS500418:8:H03J9AFXX:3:21405:9193:3657/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.89233 NS500418:8:H03J9AFXX:1:11106:16727:3933/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.89233 NS500418:8:H03J9AFXX:1:11106:16727:3933/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3484976 NS500418:8:H03J9AFXX:3:21403:8341:3072/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3484976 NS500418:8:H03J9AFXX:3:21403:8341:3072/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5693803 NS500418:8:H03J9AFXX:4:21612:24915:9386/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5693803 NS500418:8:H03J9AFXX:4:21612:24915:9386/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5462338 NS500418:8:H03J9AFXX:4:21512:17606:18242/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5462338 NS500418:8:H03J9AFXX:4:21512:17606:18242/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1349807 NS500418:8:H03J9AFXX:1:21312:23305:4786/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1349807 NS500418:8:H03J9AFXX:1:21312:23305:4786/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3479819 NS500418:8:H03J9AFXX:3:21402:24878:17458/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3479819 NS500418:8:H03J9AFXX:3:21402:24878:17458/2' because it was < 2 characters long\n",
      "99484 reads; of these:\n",
      "  99484 (100.00%) were paired; of these:\n",
      "    99122 (99.64%) aligned concordantly 0 times\n",
      "    248 (0.25%) aligned concordantly exactly 1 time\n",
      "    114 (0.11%) aligned concordantly >1 times\n",
      "    ----\n",
      "    99122 pairs aligned concordantly 0 times; of these:\n",
      "      0 (0.00%) aligned discordantly 1 time\n",
      "    ----\n",
      "    99122 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      198244 mates make up the pairs; of these:\n",
      "        197057 (99.40%) aligned 0 times\n",
      "        78 (0.04%) aligned exactly 1 time\n",
      "        1109 (0.56%) aligned >1 times\n",
      "0.96% overall alignment rate\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir -p tmp\n",
    "for i in {3,4,7,8}; \n",
    "do bowtie2 -p 4 -X 2000 --very-sensitive -x reference/yeast_MT -1 output/trimmed_files/SRR182215${i}_1_trimmed.fastq -2 output/trimmed_files/SRR182215${i}_2_trimmed.fastq -S tmp/SRR182215${i}_MT.sam --un-conc-gz tmp/SRR182215${i}_notMT.fastq.gz;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This file containing the alignment on the mitochondria won't be used, so we remove it. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\r",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm tmp/SRR182215{3,4,7,8}_MT.sam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Alignment on chromosomal genome\n",
    "\n",
    "We now align the remaining reads on the nuclear genome, in a very similar manner. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR1822153_MT.sam            SRR1822157_MT.sam\n",
      "\u001b[0m\u001b[01;31mSRR1822153_notMT_1.fastq.gz\u001b[0m  \u001b[01;31mSRR1822157_notMT_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822153_notMT_2.fastq.gz\u001b[0m  \u001b[01;31mSRR1822157_notMT_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822153_notMT.fastq.1.gz\u001b[0m  \u001b[01;31mSRR1822157_notMT.fastq.1.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822153_notMT.fastq.2.gz\u001b[0m  \u001b[01;31mSRR1822157_notMT.fastq.2.gz\u001b[0m\n",
      "SRR1822154_MT.sam            SRR1822158_MT.sam\n",
      "\u001b[01;31mSRR1822154_notMT_1.fastq.gz\u001b[0m  \u001b[01;31mSRR1822158_notMT_1.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822154_notMT_2.fastq.gz\u001b[0m  \u001b[01;31mSRR1822158_notMT_2.fastq.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822154_notMT.fastq.1.gz\u001b[0m  \u001b[01;31mSRR1822158_notMT.fastq.1.gz\u001b[0m\n",
      "\u001b[01;31mSRR1822154_notMT.fastq.2.gz\u001b[0m  \u001b[01;31mSRR1822158_notMT.fastq.2.gz\u001b[0m\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: impossible de créer le répertoire « output/aligned_files »: Le fichier existe\n",
      "3[?2004h(ATAC-Seq) \u001b[?2004l\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "Warning: skipping mate #2 of read 'SRR1822153.917955 NS500418:8:H03J9AFXX:1:21311:1438:11285/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.917955 NS500418:8:H03J9AFXX:1:21311:1438:11285/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1868166 NS500418:8:H03J9AFXX:3:11403:26762:7214/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1868166 NS500418:8:H03J9AFXX:3:11403:26762:7214/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3438736 NS500418:8:H03J9AFXX:4:21405:14687:17026/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3438736 NS500418:8:H03J9AFXX:4:21405:14687:17026/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1816668 NS500418:8:H03J9AFXX:2:21311:14382:9820/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1816668 NS500418:8:H03J9AFXX:2:21311:14382:9820/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.676592 NS500418:8:H03J9AFXX:1:21204:18996:17988/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.676592 NS500418:8:H03J9AFXX:1:21204:18996:17988/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.467856 NS500418:8:H03J9AFXX:1:21101:23818:2679/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.467856 NS500418:8:H03J9AFXX:1:21101:23818:2679/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.605522 NS500418:8:H03J9AFXX:1:21111:22965:12905/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.605522 NS500418:8:H03J9AFXX:1:21111:22965:12905/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3663238 NS500418:8:H03J9AFXX:4:21509:1551:6318/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3663238 NS500418:8:H03J9AFXX:4:21509:1551:6318/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2569466 NS500418:8:H03J9AFXX:3:21505:25326:4177/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2569466 NS500418:8:H03J9AFXX:3:21505:25326:4177/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3529729 NS500418:8:H03J9AFXX:4:21411:24743:19767/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3529729 NS500418:8:H03J9AFXX:4:21411:24743:19767/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1250538 NS500418:8:H03J9AFXX:2:11301:3744:14775/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1250538 NS500418:8:H03J9AFXX:2:11301:3744:14775/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2272841 NS500418:8:H03J9AFXX:3:11608:2975:2109/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2272841 NS500418:8:H03J9AFXX:3:11608:2975:2109/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2660323 NS500418:8:H03J9AFXX:3:21511:23369:11084/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2660323 NS500418:8:H03J9AFXX:3:21511:23369:11084/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.533129 NS500418:8:H03J9AFXX:1:21106:4356:4200/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.533129 NS500418:8:H03J9AFXX:1:21106:4356:4200/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3708432 NS500418:8:H03J9AFXX:4:21512:23237:9254/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3708432 NS500418:8:H03J9AFXX:4:21512:23237:9254/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.223800 NS500418:8:H03J9AFXX:1:11206:22327:10008/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.223800 NS500418:8:H03J9AFXX:1:11206:22327:10008/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2538328 NS500418:8:H03J9AFXX:3:21503:10909:1368/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2538328 NS500418:8:H03J9AFXX:3:21503:10909:1368/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1806093 NS500418:8:H03J9AFXX:2:21310:9636:12403/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1806093 NS500418:8:H03J9AFXX:2:21310:9636:12403/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3196257 NS500418:8:H03J9AFXX:4:11512:18377:14844/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3196257 NS500418:8:H03J9AFXX:4:11512:18377:14844/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2854196 NS500418:8:H03J9AFXX:3:21612:14487:14227/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2854196 NS500418:8:H03J9AFXX:3:21612:14487:14227/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.201769 NS500418:8:H03J9AFXX:1:11204:9726:17120/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.201769 NS500418:8:H03J9AFXX:1:11204:9726:17120/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1735879 NS500418:8:H03J9AFXX:2:21304:1721:20258/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1735879 NS500418:8:H03J9AFXX:2:21304:1721:20258/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.909146 NS500418:8:H03J9AFXX:1:21310:14603:16663/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.909146 NS500418:8:H03J9AFXX:1:21310:14603:16663/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.839160 NS500418:8:H03J9AFXX:1:21305:17462:5295/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.839160 NS500418:8:H03J9AFXX:1:21305:17462:5295/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2645730 NS500418:8:H03J9AFXX:3:21510:26677:10521/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2645730 NS500418:8:H03J9AFXX:3:21510:26677:10521/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.37567 NS500418:8:H03J9AFXX:1:11104:8888:7453/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.37567 NS500418:8:H03J9AFXX:1:11104:8888:7453/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.505219 NS500418:8:H03J9AFXX:1:21104:14594:2028/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.505219 NS500418:8:H03J9AFXX:1:21104:14594:2028/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.295973 NS500418:8:H03J9AFXX:1:11211:14264:14974/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.295973 NS500418:8:H03J9AFXX:1:11211:14264:14974/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2822565 NS500418:8:H03J9AFXX:3:21610:17626:12598/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2822565 NS500418:8:H03J9AFXX:3:21610:17626:12598/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.847550 NS500418:8:H03J9AFXX:1:21305:9231:17934/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.847550 NS500418:8:H03J9AFXX:1:21305:9231:17934/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822153.537633 NS500418:8:H03J9AFXX:1:21106:8850:11130/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.537633 NS500418:8:H03J9AFXX:1:21106:8850:11130/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2054146 NS500418:8:H03J9AFXX:3:11504:20083:14575/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2054146 NS500418:8:H03J9AFXX:3:11504:20083:14575/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3088362 NS500418:8:H03J9AFXX:4:11505:18956:5107/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3088362 NS500418:8:H03J9AFXX:4:11505:18956:5107/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3641787 NS500418:8:H03J9AFXX:4:21507:25609:16184/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3641787 NS500418:8:H03J9AFXX:4:21507:25609:16184/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3522657 NS500418:8:H03J9AFXX:4:21411:5956:10509/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3522657 NS500418:8:H03J9AFXX:4:21411:5956:10509/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.176802 NS500418:8:H03J9AFXX:1:11203:17694:1083/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.176802 NS500418:8:H03J9AFXX:1:11203:17694:1083/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.474080 NS500418:8:H03J9AFXX:1:21101:21989:12890/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.474080 NS500418:8:H03J9AFXX:1:21101:21989:12890/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.1585486 NS500418:8:H03J9AFXX:2:21205:15721:6492/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.1585486 NS500418:8:H03J9AFXX:2:21205:15721:6492/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1086033 NS500418:8:H03J9AFXX:2:11201:25427:6788/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1086033 NS500418:8:H03J9AFXX:2:11201:25427:6788/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3720432 NS500418:8:H03J9AFXX:4:21601:24867:6709/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3720432 NS500418:8:H03J9AFXX:4:21601:24867:6709/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2131036 NS500418:8:H03J9AFXX:3:11509:23890:19447/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2131036 NS500418:8:H03J9AFXX:3:11509:23890:19447/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2830271 NS500418:8:H03J9AFXX:3:21611:11249:2527/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2830271 NS500418:8:H03J9AFXX:3:21611:11249:2527/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.536918 NS500418:8:H03J9AFXX:1:21106:26351:10081/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.536918 NS500418:8:H03J9AFXX:1:21106:26351:10081/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3832705 NS500418:8:H03J9AFXX:4:21609:23511:12958/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3832705 NS500418:8:H03J9AFXX:4:21609:23511:12958/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.67053 NS500418:8:H03J9AFXX:1:11106:26037:14266/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.67053 NS500418:8:H03J9AFXX:1:11106:26037:14266/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1221778 NS500418:8:H03J9AFXX:2:11211:17153:11176/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1221778 NS500418:8:H03J9AFXX:2:11211:17153:11176/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3633885 NS500418:8:H03J9AFXX:4:21507:25189:5420/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3633885 NS500418:8:H03J9AFXX:4:21507:25189:5420/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.316883 NS500418:8:H03J9AFXX:1:11301:14152:6465/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.316883 NS500418:8:H03J9AFXX:1:11301:14152:6465/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2504426 NS500418:8:H03J9AFXX:3:21412:3654:14250/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2504426 NS500418:8:H03J9AFXX:3:21412:3654:14250/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.107818 NS500418:8:H03J9AFXX:1:11109:2948:16944/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.107818 NS500418:8:H03J9AFXX:1:11109:2948:16944/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.981689 NS500418:8:H03J9AFXX:2:11104:3645:17951/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.981689 NS500418:8:H03J9AFXX:2:11104:3645:17951/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3623996 NS500418:8:H03J9AFXX:4:21506:21993:11541/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3623996 NS500418:8:H03J9AFXX:4:21506:21993:11541/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1026570 NS500418:8:H03J9AFXX:2:11108:24898:11265/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1026570 NS500418:8:H03J9AFXX:2:11108:24898:11265/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1983053 NS500418:8:H03J9AFXX:3:11411:1110:15611/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1983053 NS500418:8:H03J9AFXX:3:11411:1110:15611/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3665431 NS500418:8:H03J9AFXX:4:21509:20222:9473/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3665431 NS500418:8:H03J9AFXX:4:21509:20222:9473/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2135201 NS500418:8:H03J9AFXX:3:11510:25853:5664/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2135201 NS500418:8:H03J9AFXX:3:11510:25853:5664/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1557499 NS500418:8:H03J9AFXX:2:21203:17716:2968/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1557499 NS500418:8:H03J9AFXX:2:21203:17716:2968/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3679704 NS500418:8:H03J9AFXX:4:21510:13451:9179/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3679704 NS500418:8:H03J9AFXX:4:21510:13451:9179/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2067676 NS500418:8:H03J9AFXX:3:11505:26301:13158/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2067676 NS500418:8:H03J9AFXX:3:11505:26301:13158/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.2458807 NS500418:8:H03J9AFXX:3:21409:22609:3969/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.2458807 NS500418:8:H03J9AFXX:3:21409:22609:3969/1' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822153.2553492 NS500418:8:H03J9AFXX:3:21504:25810:2359/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2553492 NS500418:8:H03J9AFXX:3:21504:25810:2359/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3061818 NS500418:8:H03J9AFXX:4:11503:20950:8106/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822153.3061818 NS500418:8:H03J9AFXX:4:11503:20950:8106/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3050149 NS500418:8:H03J9AFXX:4:11502:6845:11878/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3050149 NS500418:8:H03J9AFXX:4:11502:6845:11878/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.87379 NS500418:8:H03J9AFXX:1:11108:6366:6011/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.87379 NS500418:8:H03J9AFXX:1:11108:6366:6011/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1401081 NS500418:8:H03J9AFXX:2:21102:6147:5013/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1401081 NS500418:8:H03J9AFXX:2:21102:6147:5013/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3089135 NS500418:8:H03J9AFXX:4:11505:11736:6079/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3089135 NS500418:8:H03J9AFXX:4:11505:11736:6079/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1651556 NS500418:8:H03J9AFXX:2:21210:14158:8271/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.1651556 NS500418:8:H03J9AFXX:2:21210:14158:8271/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3578127 NS500418:8:H03J9AFXX:4:21503:24969:6541/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3578127 NS500418:8:H03J9AFXX:4:21503:24969:6541/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.576531 NS500418:8:H03J9AFXX:1:21109:23867:9153/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.576531 NS500418:8:H03J9AFXX:1:21109:23867:9153/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.897382 NS500418:8:H03J9AFXX:1:21309:17004:17628/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.897382 NS500418:8:H03J9AFXX:1:21309:17004:17628/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3828546 NS500418:8:H03J9AFXX:4:21609:15182:6959/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.3828546 NS500418:8:H03J9AFXX:4:21609:15182:6959/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2639655 NS500418:8:H03J9AFXX:3:21510:4530:1922/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2639655 NS500418:8:H03J9AFXX:3:21510:4530:1922/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2693670 NS500418:8:H03J9AFXX:3:21601:24058:18516/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822153.2693670 NS500418:8:H03J9AFXX:3:21601:24058:18516/2' because it was < 2 characters long\n",
      "98451 reads; of these:\n",
      "  98451 (100.00%) were paired; of these:\n",
      "    4877 (4.95%) aligned concordantly 0 times\n",
      "    48367 (49.13%) aligned concordantly exactly 1 time\n",
      "    45207 (45.92%) aligned concordantly >1 times\n",
      "    ----\n",
      "    4877 pairs aligned concordantly 0 times; of these:\n",
      "      21 (0.43%) aligned discordantly 1 time\n",
      "    ----\n",
      "    4856 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      9712 mates make up the pairs; of these:\n",
      "        8711 (89.69%) aligned 0 times\n",
      "        210 (2.16%) aligned exactly 1 time\n",
      "        791 (8.14%) aligned >1 times\n",
      "95.58% overall alignment rate\n",
      "4\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6619810 NS500418:8:H03J9AFXX:3:21512:15180:17756/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6619810 NS500418:8:H03J9AFXX:3:21512:15180:17756/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5375876 NS500418:8:H03J9AFXX:3:11601:26208:12883/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5375876 NS500418:8:H03J9AFXX:3:11601:26208:12883/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4301366 NS500418:8:H03J9AFXX:2:21306:18235:9660/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4301366 NS500418:8:H03J9AFXX:2:21306:18235:9660/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012279 NS500418:8:H03J9AFXX:4:21506:4266:19438/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012279 NS500418:8:H03J9AFXX:4:21506:4266:19438/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7297360 NS500418:8:H03J9AFXX:4:11407:25658:3755/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7297360 NS500418:8:H03J9AFXX:4:11407:25658:3755/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6797166 NS500418:8:H03J9AFXX:3:21605:12156:17787/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6797166 NS500418:8:H03J9AFXX:3:21605:12156:17787/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.912465 NS500418:8:H03J9AFXX:1:11305:11839:13985/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.912465 NS500418:8:H03J9AFXX:1:11305:11839:13985/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4743093 NS500418:8:H03J9AFXX:3:11407:5789:15822/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4743093 NS500418:8:H03J9AFXX:3:11407:5789:15822/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7305707 NS500418:8:H03J9AFXX:4:11407:1799:8175/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7305707 NS500418:8:H03J9AFXX:4:11407:1799:8175/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4874276 NS500418:8:H03J9AFXX:3:11411:24601:13205/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4874276 NS500418:8:H03J9AFXX:3:11411:24601:13205/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6069398 NS500418:8:H03J9AFXX:3:21409:17500:2979/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6069398 NS500418:8:H03J9AFXX:3:21409:17500:2979/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9506063 NS500418:8:H03J9AFXX:4:21609:18368:5721/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9506063 NS500418:8:H03J9AFXX:4:21609:18368:5721/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.7879513 NS500418:8:H03J9AFXX:4:11511:25797:10154/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.7879513 NS500418:8:H03J9AFXX:4:11511:25797:10154/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012808 NS500418:8:H03J9AFXX:4:21506:18761:19698/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.9012808 NS500418:8:H03J9AFXX:4:21506:18761:19698/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3664145 NS500418:8:H03J9AFXX:2:21110:20419:1624/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3664145 NS500418:8:H03J9AFXX:2:21110:20419:1624/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #1 of read 'SRR1822154.3513824 NS500418:8:H03J9AFXX:2:21104:2431:19125/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.3513824 NS500418:8:H03J9AFXX:2:21104:2431:19125/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7904923 NS500418:8:H03J9AFXX:4:11512:23465:4400/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7904923 NS500418:8:H03J9AFXX:4:11512:23465:4400/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5537589 NS500418:8:H03J9AFXX:3:11606:20305:6399/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5537589 NS500418:8:H03J9AFXX:3:11606:20305:6399/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.379073 NS500418:8:H03J9AFXX:1:11201:23092:9050/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.379073 NS500418:8:H03J9AFXX:1:11201:23092:9050/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4311319 NS500418:8:H03J9AFXX:2:21306:2974:15830/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4311319 NS500418:8:H03J9AFXX:2:21306:2974:15830/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2228424 NS500418:8:H03J9AFXX:1:21310:22521:19044/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2228424 NS500418:8:H03J9AFXX:1:21310:22521:19044/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.26644 NS500418:8:H03J9AFXX:1:11101:26863:20098/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.26644 NS500418:8:H03J9AFXX:1:11101:26863:20098/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2636123 NS500418:8:H03J9AFXX:2:11112:16593:10529/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2636123 NS500418:8:H03J9AFXX:2:11112:16593:10529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5940626 NS500418:8:H03J9AFXX:3:21405:19168:7424/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5940626 NS500418:8:H03J9AFXX:3:21405:19168:7424/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6726585 NS500418:8:H03J9AFXX:3:21603:23383:19601/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6726585 NS500418:8:H03J9AFXX:3:21603:23383:19601/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8514061 NS500418:8:H03J9AFXX:4:21405:7936:7343/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8514061 NS500418:8:H03J9AFXX:4:21405:7936:7343/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7869584 NS500418:8:H03J9AFXX:4:11511:21477:4857/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.7869584 NS500418:8:H03J9AFXX:4:11511:21477:4857/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.431690 NS500418:8:H03J9AFXX:1:11202:3805:20065/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.431690 NS500418:8:H03J9AFXX:1:11202:3805:20065/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.9409189 NS500418:8:H03J9AFXX:4:21606:26701:11337/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.9409189 NS500418:8:H03J9AFXX:4:21606:26701:11337/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1532495 NS500418:8:H03J9AFXX:1:21201:22051:5369/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1532495 NS500418:8:H03J9AFXX:1:21201:22051:5369/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6795534 NS500418:8:H03J9AFXX:3:21605:25638:16990/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6795534 NS500418:8:H03J9AFXX:3:21605:25638:16990/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3162043 NS500418:8:H03J9AFXX:2:11304:18127:19777/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3162043 NS500418:8:H03J9AFXX:2:11304:18127:19777/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8098985 NS500418:8:H03J9AFXX:4:11605:13396:19467/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8098985 NS500418:8:H03J9AFXX:4:11605:13396:19467/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5269966 NS500418:8:H03J9AFXX:3:11510:13634:11671/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5269966 NS500418:8:H03J9AFXX:3:11510:13634:11671/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2516065 NS500418:8:H03J9AFXX:2:11108:24219:12077/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2516065 NS500418:8:H03J9AFXX:2:11108:24219:12077/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3324702 NS500418:8:H03J9AFXX:2:11310:3535:6190/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3324702 NS500418:8:H03J9AFXX:2:11310:3535:6190/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8266923 NS500418:8:H03J9AFXX:4:11610:15558:13142/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8266923 NS500418:8:H03J9AFXX:4:11610:15558:13142/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8572612 NS500418:8:H03J9AFXX:4:21406:18591:18319/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8572612 NS500418:8:H03J9AFXX:4:21406:18591:18319/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5247786 NS500418:8:H03J9AFXX:3:11509:16612:19219/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5247786 NS500418:8:H03J9AFXX:3:11509:16612:19219/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.246332 NS500418:8:H03J9AFXX:1:11109:4901:5414/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.246332 NS500418:8:H03J9AFXX:1:11109:4901:5414/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3520675 NS500418:8:H03J9AFXX:2:21105:2345:4258/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3520675 NS500418:8:H03J9AFXX:2:21105:2345:4258/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3391032 NS500418:8:H03J9AFXX:2:11312:7456:11981/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3391032 NS500418:8:H03J9AFXX:2:11312:7456:11981/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3039879 NS500418:8:H03J9AFXX:2:11212:16967:20392/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.3039879 NS500418:8:H03J9AFXX:2:11212:16967:20392/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2357923 NS500418:8:H03J9AFXX:2:11103:18857:6625/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2357923 NS500418:8:H03J9AFXX:2:11103:18857:6625/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5878374 NS500418:8:H03J9AFXX:3:21403:7081:11334/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5878374 NS500418:8:H03J9AFXX:3:21403:7081:11334/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822154.1266814 NS500418:8:H03J9AFXX:1:21104:24432:19972/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.1266814 NS500418:8:H03J9AFXX:1:21104:24432:19972/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6106920 NS500418:8:H03J9AFXX:3:21410:10571:5689/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6106920 NS500418:8:H03J9AFXX:3:21410:10571:5689/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4187359 NS500418:8:H03J9AFXX:2:21302:23042:15949/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4187359 NS500418:8:H03J9AFXX:2:21302:23042:15949/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5608565 NS500418:8:H03J9AFXX:3:11608:22684:3902/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.5608565 NS500418:8:H03J9AFXX:3:11608:22684:3902/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2317140 NS500418:8:H03J9AFXX:2:11101:9968:18331/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2317140 NS500418:8:H03J9AFXX:2:11101:9968:18331/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2346183 NS500418:8:H03J9AFXX:2:11102:22622:18160/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.2346183 NS500418:8:H03J9AFXX:2:11102:22622:18160/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.6820003 NS500418:8:H03J9AFXX:3:21606:14057:10145/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.6820003 NS500418:8:H03J9AFXX:3:21606:14057:10145/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6690186 NS500418:8:H03J9AFXX:3:21602:17108:18974/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.6690186 NS500418:8:H03J9AFXX:3:21602:17108:18974/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8203240 NS500418:8:H03J9AFXX:4:11608:24474:17936/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.8203240 NS500418:8:H03J9AFXX:4:11608:24474:17936/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4066696 NS500418:8:H03J9AFXX:2:21210:23966:19204/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4066696 NS500418:8:H03J9AFXX:2:21210:23966:19204/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4381296 NS500418:8:H03J9AFXX:2:21309:24821:1360/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822154.4381296 NS500418:8:H03J9AFXX:2:21309:24821:1360/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822154.5491602 NS500418:8:H03J9AFXX:3:11605:5925:1859/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822154.5491602 NS500418:8:H03J9AFXX:3:11605:5925:1859/1' because it was < 2 characters long\n",
      "98497 reads; of these:\n",
      "  98497 (100.00%) were paired; of these:\n",
      "    7400 (7.51%) aligned concordantly 0 times\n",
      "    39882 (40.49%) aligned concordantly exactly 1 time\n",
      "    51215 (52.00%) aligned concordantly >1 times\n",
      "    ----\n",
      "    7400 pairs aligned concordantly 0 times; of these:\n",
      "      22 (0.30%) aligned discordantly 1 time\n",
      "    ----\n",
      "    7378 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      14756 mates make up the pairs; of these:\n",
      "        13630 (92.37%) aligned 0 times\n",
      "        211 (1.43%) aligned exactly 1 time\n",
      "        915 (6.20%) aligned >1 times\n",
      "93.08% overall alignment rate\n",
      "7\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4244520 NS500418:8:H03J9AFXX:4:21404:13861:10491/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4244520 NS500418:8:H03J9AFXX:4:21404:13861:10491/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874583 NS500418:8:H03J9AFXX:1:21207:11553:10798/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874583 NS500418:8:H03J9AFXX:1:21207:11553:10798/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2403542 NS500418:8:H03J9AFXX:3:11409:22143:3243/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2403542 NS500418:8:H03J9AFXX:3:11409:22143:3243/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1235169 NS500418:8:H03J9AFXX:2:11106:25548:14740/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1235169 NS500418:8:H03J9AFXX:2:11106:25548:14740/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.194140 NS500418:8:H03J9AFXX:1:11201:21745:11469/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.194140 NS500418:8:H03J9AFXX:1:11201:21745:11469/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874842 NS500418:8:H03J9AFXX:1:21207:6471:11082/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.874842 NS500418:8:H03J9AFXX:1:21207:6471:11082/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4466261 NS500418:8:H03J9AFXX:4:21504:14323:15574/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4466261 NS500418:8:H03J9AFXX:4:21504:14323:15574/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822157.1043285 NS500418:8:H03J9AFXX:1:21306:22089:1077/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822157.1043285 NS500418:8:H03J9AFXX:1:21306:22089:1077/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3558482 NS500418:8:H03J9AFXX:4:11401:7740:17153/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3558482 NS500418:8:H03J9AFXX:4:11401:7740:17153/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4291071 NS500418:8:H03J9AFXX:4:21406:24053:20054/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4291071 NS500418:8:H03J9AFXX:4:21406:24053:20054/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3155580 NS500418:8:H03J9AFXX:3:21503:23507:13435/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3155580 NS500418:8:H03J9AFXX:3:21503:23507:13435/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1952138 NS500418:8:H03J9AFXX:2:21205:2736:4863/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1952138 NS500418:8:H03J9AFXX:2:21205:2736:4863/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4566635 NS500418:8:H03J9AFXX:4:21510:7902:9560/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4566635 NS500418:8:H03J9AFXX:4:21510:7902:9560/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2596222 NS500418:8:H03J9AFXX:3:11507:19465:18428/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2596222 NS500418:8:H03J9AFXX:3:11507:19465:18428/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2469249 NS500418:8:H03J9AFXX:3:11501:4623:2573/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2469249 NS500418:8:H03J9AFXX:3:11501:4623:2573/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1947513 NS500418:8:H03J9AFXX:2:21204:21559:18573/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1947513 NS500418:8:H03J9AFXX:2:21204:21559:18573/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822157.4778970 NS500418:8:H03J9AFXX:4:21610:4469:17529/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4778970 NS500418:8:H03J9AFXX:4:21610:4469:17529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3517948 NS500418:8:H03J9AFXX:3:21611:25487:14102/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3517948 NS500418:8:H03J9AFXX:3:21611:25487:14102/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2940338 NS500418:8:H03J9AFXX:3:21402:1485:19745/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2940338 NS500418:8:H03J9AFXX:3:21402:1485:19745/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4022474 NS500418:8:H03J9AFXX:4:11604:10254:3180/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4022474 NS500418:8:H03J9AFXX:4:11604:10254:3180/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1964597 NS500418:8:H03J9AFXX:2:21205:5166:19904/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1964597 NS500418:8:H03J9AFXX:2:21205:5166:19904/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.246175 NS500418:8:H03J9AFXX:1:11204:17318:13066/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.246175 NS500418:8:H03J9AFXX:1:11204:17318:13066/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3571863 NS500418:8:H03J9AFXX:4:11402:15222:11807/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3571863 NS500418:8:H03J9AFXX:4:11402:15222:11807/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4707532 NS500418:8:H03J9AFXX:4:21606:2744:17315/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4707532 NS500418:8:H03J9AFXX:4:21606:2744:17315/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3589098 NS500418:8:H03J9AFXX:4:11403:21355:10333/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3589098 NS500418:8:H03J9AFXX:4:11403:21355:10333/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.579916 NS500418:8:H03J9AFXX:1:21101:1692:4498/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.579916 NS500418:8:H03J9AFXX:1:21101:1692:4498/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2643943 NS500418:8:H03J9AFXX:3:11510:5522:11322/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2643943 NS500418:8:H03J9AFXX:3:11510:5522:11322/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.311609 NS500418:8:H03J9AFXX:1:11208:7519:11193/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.311609 NS500418:8:H03J9AFXX:1:11208:7519:11193/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3568179 NS500418:8:H03J9AFXX:4:11402:9141:7864/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3568179 NS500418:8:H03J9AFXX:4:11402:9141:7864/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1527361 NS500418:8:H03J9AFXX:2:11212:18527:20335/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1527361 NS500418:8:H03J9AFXX:2:11212:18527:20335/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790589 NS500418:8:H03J9AFXX:4:11502:23067:16911/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790589 NS500418:8:H03J9AFXX:4:11502:23067:16911/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4305628 NS500418:8:H03J9AFXX:4:21407:26618:16283/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4305628 NS500418:8:H03J9AFXX:4:21407:26618:16283/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4564163 NS500418:8:H03J9AFXX:4:21510:14511:6815/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4564163 NS500418:8:H03J9AFXX:4:21510:14511:6815/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1424342 NS500418:8:H03J9AFXX:2:11206:21685:13980/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1424342 NS500418:8:H03J9AFXX:2:11206:21685:13980/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.602210 NS500418:8:H03J9AFXX:1:21102:20234:13842/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.602210 NS500418:8:H03J9AFXX:1:21102:20234:13842/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4272542 NS500418:8:H03J9AFXX:4:21405:5930:20354/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4272542 NS500418:8:H03J9AFXX:4:21405:5930:20354/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2869676 NS500418:8:H03J9AFXX:3:11611:7632:1955/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2869676 NS500418:8:H03J9AFXX:3:11611:7632:1955/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2896384 NS500418:8:H03J9AFXX:3:11612:20992:9899/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2896384 NS500418:8:H03J9AFXX:3:11612:20992:9899/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1606448 NS500418:8:H03J9AFXX:2:11306:3168:2556/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1606448 NS500418:8:H03J9AFXX:2:11306:3168:2556/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4449449 NS500418:8:H03J9AFXX:4:21503:1093:15969/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4449449 NS500418:8:H03J9AFXX:4:21503:1093:15969/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2650855 NS500418:8:H03J9AFXX:3:11510:12270:18842/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2650855 NS500418:8:H03J9AFXX:3:11510:12270:18842/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2012581 NS500418:8:H03J9AFXX:2:21208:12294:20282/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2012581 NS500418:8:H03J9AFXX:2:21208:12294:20282/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1135462 NS500418:8:H03J9AFXX:1:21311:15773:20238/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1135462 NS500418:8:H03J9AFXX:1:21311:15773:20238/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.182673 NS500418:8:H03J9AFXX:1:11112:3439:17433/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.182673 NS500418:8:H03J9AFXX:1:11112:3439:17433/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1101534 NS500418:8:H03J9AFXX:1:21309:26518:15716/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1101534 NS500418:8:H03J9AFXX:1:21309:26518:15716/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1291566 NS500418:8:H03J9AFXX:2:11110:19382:8350/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1291566 NS500418:8:H03J9AFXX:2:11110:19382:8350/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822157.3436796 NS500418:8:H03J9AFXX:3:21607:4534:8800/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3436796 NS500418:8:H03J9AFXX:3:21607:4534:8800/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2533842 NS500418:8:H03J9AFXX:3:11504:13926:11277/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2533842 NS500418:8:H03J9AFXX:3:11504:13926:11277/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3260827 NS500418:8:H03J9AFXX:3:21509:23447:11138/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3260827 NS500418:8:H03J9AFXX:3:21509:23447:11138/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790923 NS500418:8:H03J9AFXX:4:11502:2233:17239/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790923 NS500418:8:H03J9AFXX:4:11502:2233:17239/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2683443 NS500418:8:H03J9AFXX:3:11512:23587:15558/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.2683443 NS500418:8:H03J9AFXX:3:11512:23587:15558/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3927746 NS500418:8:H03J9AFXX:4:11510:4583:10311/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3927746 NS500418:8:H03J9AFXX:4:11510:4583:10311/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.733070 NS500418:8:H03J9AFXX:1:21110:1441:18600/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.733070 NS500418:8:H03J9AFXX:1:21110:1441:18600/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4570459 NS500418:8:H03J9AFXX:4:21510:7398:13841/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4570459 NS500418:8:H03J9AFXX:4:21510:7398:13841/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4615013 NS500418:8:H03J9AFXX:4:21601:22050:4559/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4615013 NS500418:8:H03J9AFXX:4:21601:22050:4559/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.905288 NS500418:8:H03J9AFXX:1:21209:3978:8004/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.905288 NS500418:8:H03J9AFXX:1:21209:3978:8004/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4149075 NS500418:8:H03J9AFXX:4:11611:4775:5530/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.4149075 NS500418:8:H03J9AFXX:4:11611:4775:5530/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.893037 NS500418:8:H03J9AFXX:1:21208:18960:13181/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.893037 NS500418:8:H03J9AFXX:1:21208:18960:13181/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.763997 NS500418:8:H03J9AFXX:1:21112:9756:17010/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.763997 NS500418:8:H03J9AFXX:1:21112:9756:17010/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.139747 NS500418:8:H03J9AFXX:1:11110:24875:4331/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.139747 NS500418:8:H03J9AFXX:1:11110:24875:4331/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790514 NS500418:8:H03J9AFXX:4:11502:10204:16838/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.3790514 NS500418:8:H03J9AFXX:4:11502:10204:16838/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1432839 NS500418:8:H03J9AFXX:2:11207:5542:4960/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1432839 NS500418:8:H03J9AFXX:2:11207:5542:4960/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.663627 NS500418:8:H03J9AFXX:1:21106:22227:12106/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.663627 NS500418:8:H03J9AFXX:1:21106:22227:12106/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1737749 NS500418:8:H03J9AFXX:2:21102:5051:19318/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.1737749 NS500418:8:H03J9AFXX:2:21102:5051:19318/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822157.922149 NS500418:8:H03J9AFXX:1:21210:6982:8403/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822157.922149 NS500418:8:H03J9AFXX:1:21210:6982:8403/2' because it was < 2 characters long\n",
      "99252 reads; of these:\n",
      "  99252 (100.00%) were paired; of these:\n",
      "    5615 (5.66%) aligned concordantly 0 times\n",
      "    59187 (59.63%) aligned concordantly exactly 1 time\n",
      "    34450 (34.71%) aligned concordantly >1 times\n",
      "    ----\n",
      "    5615 pairs aligned concordantly 0 times; of these:\n",
      "      17 (0.30%) aligned discordantly 1 time\n",
      "    ----\n",
      "    5598 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      11196 mates make up the pairs; of these:\n",
      "        10422 (93.09%) aligned 0 times\n",
      "        216 (1.93%) aligned exactly 1 time\n",
      "        558 (4.98%) aligned >1 times\n",
      "94.75% overall alignment rate\n",
      "8\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1515577 NS500418:8:H03J9AFXX:2:11109:2946:11317/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1515577 NS500418:8:H03J9AFXX:2:11109:2946:11317/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2841246 NS500418:8:H03J9AFXX:3:11408:13800:14719/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2841246 NS500418:8:H03J9AFXX:3:11408:13800:14719/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5213860 NS500418:8:H03J9AFXX:4:21412:25584:19892/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5213860 NS500418:8:H03J9AFXX:4:21412:25584:19892/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1574134 NS500418:8:H03J9AFXX:2:11112:25956:15203/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1574134 NS500418:8:H03J9AFXX:2:11112:25956:15203/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2722199 NS500418:8:H03J9AFXX:3:11402:24229:18703/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2722199 NS500418:8:H03J9AFXX:3:11402:24229:18703/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4729890 NS500418:8:H03J9AFXX:4:11602:19415:7772/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4729890 NS500418:8:H03J9AFXX:4:11602:19415:7772/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2223045 NS500418:8:H03J9AFXX:2:21112:9293:5545/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2223045 NS500418:8:H03J9AFXX:2:21112:9293:5545/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3992264 NS500418:8:H03J9AFXX:3:21603:23252:19490/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3992264 NS500418:8:H03J9AFXX:3:21603:23252:19490/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5317270 NS500418:8:H03J9AFXX:4:21505:25390:19471/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5317270 NS500418:8:H03J9AFXX:4:21505:25390:19471/2' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.3441264 NS500418:8:H03J9AFXX:3:11612:25272:19343/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3441264 NS500418:8:H03J9AFXX:3:11612:25272:19343/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4252580 NS500418:8:H03J9AFXX:4:11403:8072:14570/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4252580 NS500418:8:H03J9AFXX:4:11403:8072:14570/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5316087 NS500418:8:H03J9AFXX:4:21505:10289:18442/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5316087 NS500418:8:H03J9AFXX:4:21505:10289:18442/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3416966 NS500418:8:H03J9AFXX:3:11611:20418:17695/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3416966 NS500418:8:H03J9AFXX:3:11611:20418:17695/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1336914 NS500418:8:H03J9AFXX:1:21311:2806:10501/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1336914 NS500418:8:H03J9AFXX:1:21311:2806:10501/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2626174 NS500418:8:H03J9AFXX:2:21309:20094:19895/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2626174 NS500418:8:H03J9AFXX:2:21309:20094:19895/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5336362 NS500418:8:H03J9AFXX:4:21506:14405:18104/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5336362 NS500418:8:H03J9AFXX:4:21506:14405:18104/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4668943 NS500418:8:H03J9AFXX:4:11511:1694:7553/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4668943 NS500418:8:H03J9AFXX:4:11511:1694:7553/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4371848 NS500418:8:H03J9AFXX:4:11409:1882:4065/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4371848 NS500418:8:H03J9AFXX:4:11409:1882:4065/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5099006 NS500418:8:H03J9AFXX:4:21407:1428:15726/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5099006 NS500418:8:H03J9AFXX:4:21407:1428:15726/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.707160 NS500418:8:H03J9AFXX:1:21102:6801:6912/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.707160 NS500418:8:H03J9AFXX:1:21102:6801:6912/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4187963 NS500418:8:H03J9AFXX:3:21612:9346:16158/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4187963 NS500418:8:H03J9AFXX:3:21612:9346:16158/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4121224 NS500418:8:H03J9AFXX:3:21609:4528:15928/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4121224 NS500418:8:H03J9AFXX:3:21609:4528:15928/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4495875 NS500418:8:H03J9AFXX:4:11503:13529:4865/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4495875 NS500418:8:H03J9AFXX:4:11503:13529:4865/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2235112 NS500418:8:H03J9AFXX:2:21112:15560:18732/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2235112 NS500418:8:H03J9AFXX:2:21112:15560:18732/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142948 NS500418:8:H03J9AFXX:1:11108:23289:20264/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142948 NS500418:8:H03J9AFXX:1:11108:23289:20264/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.124102 NS500418:8:H03J9AFXX:1:11108:21213:1397/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.124102 NS500418:8:H03J9AFXX:1:11108:21213:1397/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2300328 NS500418:8:H03J9AFXX:2:21204:15766:9026/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2300328 NS500418:8:H03J9AFXX:2:21204:15766:9026/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1825373 NS500418:8:H03J9AFXX:2:11301:25074:14403/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.1825373 NS500418:8:H03J9AFXX:2:11301:25074:14403/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4683494 NS500418:8:H03J9AFXX:4:11512:4340:1487/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4683494 NS500418:8:H03J9AFXX:4:11512:4340:1487/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1651720 NS500418:8:H03J9AFXX:2:11204:3330:15570/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1651720 NS500418:8:H03J9AFXX:2:11204:3330:15570/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3875592 NS500418:8:H03J9AFXX:3:21510:11640:5180/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3875592 NS500418:8:H03J9AFXX:3:21510:11640:5180/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2540914 NS500418:8:H03J9AFXX:2:21305:20539:7658/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2540914 NS500418:8:H03J9AFXX:2:21305:20539:7658/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4779982 NS500418:8:H03J9AFXX:4:11604:5735:18480/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4779982 NS500418:8:H03J9AFXX:4:11604:5735:18480/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5674531 NS500418:8:H03J9AFXX:4:21611:25817:10879/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5674531 NS500418:8:H03J9AFXX:4:21611:25817:10879/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3829550 NS500418:8:H03J9AFXX:3:21507:16566:20315/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3829550 NS500418:8:H03J9AFXX:3:21507:16566:20315/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3455197 NS500418:8:H03J9AFXX:3:21401:14672:13422/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3455197 NS500418:8:H03J9AFXX:3:21401:14672:13422/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4199278 NS500418:8:H03J9AFXX:4:11401:10874:6406/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4199278 NS500418:8:H03J9AFXX:4:11401:10874:6406/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2425736 NS500418:8:H03J9AFXX:2:21211:7354:2340/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.2425736 NS500418:8:H03J9AFXX:2:21211:7354:2340/1' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5030491 NS500418:8:H03J9AFXX:4:21404:21612:13859/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.5030491 NS500418:8:H03J9AFXX:4:21404:21612:13859/1' because it was < 2 characters long\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.2586653 NS500418:8:H03J9AFXX:2:21307:24676:17042/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2586653 NS500418:8:H03J9AFXX:2:21307:24676:17042/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5295901 NS500418:8:H03J9AFXX:4:21504:5540:19050/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5295901 NS500418:8:H03J9AFXX:4:21504:5540:19050/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4674937 NS500418:8:H03J9AFXX:4:11511:2046:13054/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4674937 NS500418:8:H03J9AFXX:4:11511:2046:13054/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3885769 NS500418:8:H03J9AFXX:3:21510:10358:14634/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3885769 NS500418:8:H03J9AFXX:3:21510:10358:14634/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2834588 NS500418:8:H03J9AFXX:3:11408:6511:8304/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2834588 NS500418:8:H03J9AFXX:3:11408:6511:8304/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5661358 NS500418:8:H03J9AFXX:4:21610:22171:18112/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5661358 NS500418:8:H03J9AFXX:4:21610:22171:18112/2' because it was < 2 characters long\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4943824 NS500418:8:H03J9AFXX:4:11612:5472:14299/1' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #1 of read 'SRR1822158.4943824 NS500418:8:H03J9AFXX:4:11612:5472:14299/1' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4177262 NS500418:8:H03J9AFXX:3:21612:15044:6655/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4177262 NS500418:8:H03J9AFXX:3:21612:15044:6655/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2387990 NS500418:8:H03J9AFXX:2:21209:6043:1941/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2387990 NS500418:8:H03J9AFXX:2:21209:6043:1941/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1171651 NS500418:8:H03J9AFXX:1:21302:25629:9670/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1171651 NS500418:8:H03J9AFXX:1:21302:25629:9670/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4010360 NS500418:8:H03J9AFXX:3:21604:24772:16656/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4010360 NS500418:8:H03J9AFXX:3:21604:24772:16656/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4538030 NS500418:8:H03J9AFXX:4:11505:10225:4596/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4538030 NS500418:8:H03J9AFXX:4:11505:10225:4596/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4974103 NS500418:8:H03J9AFXX:4:21402:15190:2856/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4974103 NS500418:8:H03J9AFXX:4:21402:15190:2856/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2637703 NS500418:8:H03J9AFXX:2:21310:9675:12704/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2637703 NS500418:8:H03J9AFXX:2:21310:9675:12704/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.126781 NS500418:8:H03J9AFXX:1:11108:23709:4021/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.126781 NS500418:8:H03J9AFXX:1:11108:23709:4021/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1775038 NS500418:8:H03J9AFXX:2:11211:6688:2776/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1775038 NS500418:8:H03J9AFXX:2:11211:6688:2776/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5392497 NS500418:8:H03J9AFXX:4:21509:9769:12463/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5392497 NS500418:8:H03J9AFXX:4:21509:9769:12463/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2412801 NS500418:8:H03J9AFXX:2:21210:13015:8588/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2412801 NS500418:8:H03J9AFXX:2:21210:13015:8588/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1829945 NS500418:8:H03J9AFXX:2:11301:12293:19483/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1829945 NS500418:8:H03J9AFXX:2:11301:12293:19483/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5546313 NS500418:8:H03J9AFXX:4:21605:2268:7223/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5546313 NS500418:8:H03J9AFXX:4:21605:2268:7223/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3099064 NS500418:8:H03J9AFXX:3:11508:1992:19314/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3099064 NS500418:8:H03J9AFXX:3:11508:1992:19314/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1617141 NS500418:8:H03J9AFXX:2:11202:9871:19780/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1617141 NS500418:8:H03J9AFXX:2:11202:9871:19780/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.209555 NS500418:8:H03J9AFXX:1:11112:6932:10529/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.209555 NS500418:8:H03J9AFXX:1:11112:6932:10529/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.242207 NS500418:8:H03J9AFXX:1:11202:13972:4073/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.242207 NS500418:8:H03J9AFXX:1:11202:13972:4073/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.229514 NS500418:8:H03J9AFXX:1:11201:2447:11165/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.229514 NS500418:8:H03J9AFXX:1:11201:2447:11165/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3682078 NS500418:8:H03J9AFXX:3:21501:10891:2659/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3682078 NS500418:8:H03J9AFXX:3:21501:10891:2659/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3694129 NS500418:8:H03J9AFXX:3:21501:13111:14038/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3694129 NS500418:8:H03J9AFXX:3:21501:13111:14038/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2260158 NS500418:8:H03J9AFXX:2:21202:1137:6114/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.2260158 NS500418:8:H03J9AFXX:2:21202:1137:6114/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142644 NS500418:8:H03J9AFXX:1:11108:26702:19961/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.142644 NS500418:8:H03J9AFXX:1:11108:26702:19961/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3227533 NS500418:8:H03J9AFXX:3:11603:4656:6322/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3227533 NS500418:8:H03J9AFXX:3:11603:4656:6322/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.4577305 NS500418:8:H03J9AFXX:4:11507:24943:2184/2' because length (1) <= # seed mismatches (0)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning: skipping mate #2 of read 'SRR1822158.4577305 NS500418:8:H03J9AFXX:4:11507:24943:2184/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3525986 NS500418:8:H03J9AFXX:3:21405:9193:3657/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3525986 NS500418:8:H03J9AFXX:3:21405:9193:3657/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.89233 NS500418:8:H03J9AFXX:1:11106:16727:3933/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.89233 NS500418:8:H03J9AFXX:1:11106:16727:3933/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3484976 NS500418:8:H03J9AFXX:3:21403:8341:3072/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3484976 NS500418:8:H03J9AFXX:3:21403:8341:3072/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5693803 NS500418:8:H03J9AFXX:4:21612:24915:9386/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5693803 NS500418:8:H03J9AFXX:4:21612:24915:9386/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5462338 NS500418:8:H03J9AFXX:4:21512:17606:18242/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.5462338 NS500418:8:H03J9AFXX:4:21512:17606:18242/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1349807 NS500418:8:H03J9AFXX:1:21312:23305:4786/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.1349807 NS500418:8:H03J9AFXX:1:21312:23305:4786/2' because it was < 2 characters long\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3479819 NS500418:8:H03J9AFXX:3:21402:24878:17458/2' because length (1) <= # seed mismatches (0)\n",
      "Warning: skipping mate #2 of read 'SRR1822158.3479819 NS500418:8:H03J9AFXX:3:21402:24878:17458/2' because it was < 2 characters long\n",
      "99122 reads; of these:\n",
      "  99122 (100.00%) were paired; of these:\n",
      "    5780 (5.83%) aligned concordantly 0 times\n",
      "    60336 (60.87%) aligned concordantly exactly 1 time\n",
      "    33006 (33.30%) aligned concordantly >1 times\n",
      "    ----\n",
      "    5780 pairs aligned concordantly 0 times; of these:\n",
      "      25 (0.43%) aligned discordantly 1 time\n",
      "    ----\n",
      "    5755 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      11510 mates make up the pairs; of these:\n",
      "        10564 (91.78%) aligned 0 times\n",
      "        249 (2.16%) aligned exactly 1 time\n",
      "        697 (6.06%) aligned >1 times\n",
      "94.67% overall alignment rate\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir output/aligned_files\n",
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   bowtie2 -p 4 -X 2000 --very-sensitive -x reference/yeast_NC -1 tmp/SRR182215${i}_notMT.fastq.1.gz -2 tmp/SRR182215${i}_notMT.fastq.2.gz -S output/aligned_files/SRR182215${i}_notMT_aligned.sam;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We remove temporary files. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\r",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "rm -fr tmp/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We now sort the sam file, convert it to bam, and then index it."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR1822153_notMT_aligned.sam  SRR1822157_notMT_aligned.sam\n",
      "SRR1822154_notMT_aligned.sam  SRR1822158_notMT_aligned.sam\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls output/aligned_files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3[?2004h\u001b[?2004l\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   samtools sort output/aligned_files/SRR182215${i}_notMT_aligned.sam -o output/aligned_files/SRR182215${i}_notMT_aligned_sorted.bam;\n",
    "   samtools index output/aligned_files/SRR182215${i}_notMT_aligned_sorted.bam;\n",
    "   done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SRR1822153_notMT_aligned.sam\n",
      "SRR1822153_notMT_aligned_sorted.bam\n",
      "SRR1822153_notMT_aligned_sorted.bam.bai\n",
      "SRR1822154_notMT_aligned.sam\n",
      "SRR1822154_notMT_aligned_sorted.bam\n",
      "SRR1822154_notMT_aligned_sorted.bam.bai\n",
      "SRR1822157_notMT_aligned.sam\n",
      "SRR1822157_notMT_aligned_sorted.bam\n",
      "SRR1822157_notMT_aligned_sorted.bam.bai\n",
      "SRR1822158_notMT_aligned.sam\n",
      "SRR1822158_notMT_aligned_sorted.bam\n",
      "SRR1822158_notMT_aligned_sorted.bam.bai\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "ls output/aligned_files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Quality Controls and Filters\n",
    "\n",
    "The various outputs of this part, located in `output/stats/` must be studied to see whether something has gone wrong with the alignment or previous steps. What to do when you have an unexpected behavior will very much depend on your experiment.\n",
    "\n",
    "### SAMtools flags\n",
    "SAMtools flags give several information about a read, such as whether it is paired or not. Here we check the distribution of those flags within our sample.\n",
    "\n",
    "The [wikipedia page](https://en.wikipedia.org/wiki/SAM_(file_format)) about SAM files does explicit the meaning of each SAMtools flag."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3[?2004h\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   samtools flagstat output/aligned_files/SRR182215${i}_notMT_aligned_sorted.bam > output/stats/SRR182215${i}_aligned_samflags.txt;\n",
    "   done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Samtools MAPQ score\n",
    "The MAPQ score is related to the probability that a particular read is misaligned. The higher the MAPQ score is, the lower the risk that the read is misaligned is. In our experiment we used a particular threshold of 22, corresponding to a probability of misalignment of 10<sup>-12</sup>. \n",
    "Let's create the MAPQ score distribution using SAMtools for reads that have the flag that corresponds to 2 (each segment properly aligned according to aligner)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3[?2004h\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   samtools view -f2 output/aligned_files/SRR182215${i}_notMT_aligned_sorted.bam | awk '{print $5}' | sort -n | uniq -c | sed 's/^ *//g'  > output/stats/SRR182215${i}_aligned_aligned_mapq.txt;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using Bowtie2, the reads that mapped several times have a MAPQ score of 1. The reads with MAPQ > 22 are uniquely mapped with good quality. We will now filter the reads to keep only these ones.   \n",
    "\n",
    "### Filter base on the quality scores threshold you choose"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3[?2004h\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   samtools view -f2 -q22 -b output/aligned_files/SRR182215${i}_notMT_aligned_sorted.bam > output/aligned_files/SRR182215${i}_notMT_aligned_filtered.bam;\n",
    "   done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Template length\n",
    "\n",
    "Let's observe the Template Length. This is the distance between the mapped end of the DNA fragment and the mapped start of the DNA fragment, inclusively. If the mapping is correct, it corresponds to the size of the original DNA fragment. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3[?2004h\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   samtools view output/aligned_files/SRR182215${i}_notMT_aligned_filtered.bam | cut -f 9| sed 's/^-//' | sort -n | uniq -c |sed  's/^ *//g' > output/stats/SRR182215${i}_aligned_length.txt;\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### fastQC - general quality check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 5% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 10% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 15% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 20% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 25% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 30% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 35% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 40% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 45% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 50% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 55% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 60% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 65% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 70% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 75% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 80% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 85% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 90% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 95% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Approx 100% complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Analysis complete for SRR1822153_notMT_aligned_filtered.bam\n",
      "Started analysis of SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 5% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 10% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 15% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 20% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 25% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 30% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 35% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 40% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 45% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 50% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 55% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 60% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 65% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 70% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 75% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 80% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 85% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 90% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 95% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Approx 100% complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Analysis complete for SRR1822154_notMT_aligned_filtered.bam\n",
      "Started analysis of SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 5% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 10% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 15% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 20% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 25% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 30% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 35% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 40% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 45% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 50% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 55% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 60% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 65% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 70% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 75% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 80% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 85% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 90% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 95% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Approx 100% complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Analysis complete for SRR1822157_notMT_aligned_filtered.bam\n",
      "Started analysis of SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 5% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 10% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 15% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 20% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 25% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 30% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 35% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 40% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 45% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 50% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 55% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 60% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 65% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 70% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 75% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 80% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 85% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 90% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 95% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Approx 100% complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "Analysis complete for SRR1822158_notMT_aligned_filtered.bam\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir output/stats/fastQC/\n",
    "fastqc -o output/stats/fastQC/ output/aligned_files/SRR182215{3,4,7,8}_notMT_aligned_filtered.bam"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Same as before, if you are treating several files at once, multiQC can be a good way to check whether the quality vary depending on each sample/condition, which might affect the results later on."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=546198;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ 🔍 \u001b[2m| v1.11\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/mag/ATACseq/TestJupyterNotebook/ATAC-seq/output/stats/fastQC\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[38;2;114;156;31m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m8/8\u001b[0m    \n",
      "\u001b[?25h\u001b[34m|            fastqc\u001b[0m | Found 4 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : output/stats/fastQC/multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : output/stats/fastQC/multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "multiqc output/stats/fastQC/ -o output/stats/fastQC/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Peak calling\n",
    "\n",
    "To call the ATAC-Seq peaks, we use MACS2. It might be more efficient to pool the reads from similar samples. You can do it by giving several bam files when calling MACS2. If you have several conditions that you want to compare, keep them separated and create one peak file per condition.\n",
    "\n",
    "This part might need to be adapted to your particular case, so let's break down the various arguments : \n",
    "* `-t output/aligned_filtered/SRR182215{3,4,7,8}_notMT_aligned_filtered.bam` name of the file (or files) on which to do the peak calling. If you input several files, they will be pooled to give a single callpeak file.\n",
    "* `-f BAM` Type of the output file desired\n",
    "* `-g 1.2e7` Effective length of the genome (S.cerevisiae here). This has to be adapted to your genome (some are precompiled : hs: 2.7e9; mm: 1.87e9; ce: 9e7; dm: 1.2e8)\n",
    "* `-q 0.05` minimum q-value for report of the peaks\n",
    "* `-n output/ATAC_Seq_peaks/SRR182215{3,4,7,8}` base name of the output, several files containing a variety results will be generated\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO  @ Mon, 27 Sep 2021 14:52:08: \n",
      "# Command line: callpeak -t output/aligned_files/SRR1822153_notMT_aligned_filtered.bam output/aligned_files/SRR1822154_notMT_aligned_filtered.bam output/aligned_files/SRR1822157_notMT_aligned_filtered.bam output/aligned_files/SRR1822158_notMT_aligned_filtered.bam -f BAMPE -g 1.2e7 -q 0.05 -n output/ATAC_Seq_peaks/All_samples\n",
      "# ARGUMENTS LIST:\n",
      "# name = output/ATAC_Seq_peaks/All_samples\n",
      "# format = BAMPE\n",
      "# ChIP-seq file = ['output/aligned_files/SRR1822153_notMT_aligned_filtered.bam', 'output/aligned_files/SRR1822154_notMT_aligned_filtered.bam', 'output/aligned_files/SRR1822157_notMT_aligned_filtered.bam', 'output/aligned_files/SRR1822158_notMT_aligned_filtered.bam']\n",
      "# control file = None\n",
      "# effective genome size = 1.20e+07\n",
      "# band width = 300\n",
      "# model fold = [5, 50]\n",
      "# qvalue cutoff = 5.00e-02\n",
      "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
      "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
      "# Larger dataset will be scaled towards smaller dataset.\n",
      "# Range for calculating regional lambda is: 10000 bps\n",
      "# Broad region calling is off\n",
      "# Paired-End mode is on\n",
      " \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:08: #1 read fragment files... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:08: #1 read treatment fragments... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: 50172 fragments have been read. \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: 41539 fragments have been read. \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: 60389 fragments have been read. \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: 61564 fragments have been read. \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1 mean fragment size is determined as 175.5 bp from treatment \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1 fragment size = 175.5 \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1  total fragments in treatment: 213664 \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1 user defined the maximum fragments... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1 filter out redundant fragments by allowing at most 1 identical fragment(s) \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1  fragments after filtering in treatment: 209304 \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1  Redundant rate of treatment: 0.02 \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #1 finished! \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #2 Build Peak Model... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #2 Skipped... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #3 Call peaks... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:09: #3 Pre-compute pvalue-qvalue table... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:10: #3 Call peaks for each chromosome... \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:10: #4 Write output xls file... output/ATAC_Seq_peaks/All_samples_peaks.xls \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:10: #4 Write peak in narrowPeak format file... output/ATAC_Seq_peaks/All_samples_peaks.narrowPeak \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:10: #4 Write summits bed file... output/ATAC_Seq_peaks/All_samples_summits.bed \n",
      "INFO  @ Mon, 27 Sep 2021 14:52:10: Done! \n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "# I create one files with the peaks from all the samples. \n",
    "mkdir -p output/ATAC_Seq_peaks\n",
    "macs2 callpeak -t output/aligned_files/SRR182215{3,4,7,8}_notMT_aligned_filtered.bam -f BAMPE -g 1.2e7 -q 0.05 -n output/ATAC_Seq_peaks/All_samples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Mask regions\n",
    "\n",
    "Some peaks might be in regions known to create artefactual peaks. It is important to remove the corresponding peaks from the results. \n",
    "\n",
    "The blacklists can be found [here](http://mitra.stanford.edu/kundaje/akundaje/release/blacklists/) or from [ENCODE](https://github.com/Boyle-Lab/Blacklist/tree/master/lists) and should be adapted to your project. In yeast, there is no such region and we only remove rDNA peaks.  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\r",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bedtools subtract -A -a output/ATAC_Seq_peaks/All_samples_summits.bed -b data/blacklist/sacCer3_rDNA.bed > output/ATAC_Seq_peaks/All_samples_summits_BL.bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[?2004l\r",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "bedtools subtract -A -a output/ATAC_Seq_peaks/All_samples_peaks.narrowPeak -b data/blacklist/sacCer3_rDNA.bed > output/ATAC_Seq_peaks/All_samples_peaks_BL.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Peak coverage per sample\n",
    "\n",
    "We use bedtools coverage to get the peak coverage of every sample. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: impossible de créer le répertoire « output/peak_coverage/ »: Le fichier existe\n",
      "3[?2004h(ATAC-Seq) \u001b[?2004l\u001b[?2004l\u001b[?2004l\u001b[?2004l\n",
      "4\n",
      "7\n",
      "8\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "mkdir output/peak_coverage/\n",
    "for i in {3,4,7,8}; \n",
    "do echo ${i};\n",
    "   bedtools coverage -a output/ATAC_Seq_peaks/All_samples_peaks_BL.bed -b output/aligned_files/SRR182215${i}_notMT_aligned_filtered.bam > output/peak_coverage/SRR182215${i}_peaks.bed;\n",
    "   done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Those outputs can be used to run differential analysis between 2 conditions with DEseq2 or edgeR. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "I\t38\t709\toutput/ATAC_Seq_peaks/All_samples_peak_1\t489\t.\t12.4987\t52.8872\t48.9184\t164\t157\t671\t671\t1.0000000\n",
      "I\t20734\t21103\toutput/ATAC_Seq_peaks/All_samples_peak_2\t92\t.\t5.21553\t11.216\t9.28764\t192\t20\t369\t369\t1.0000000\n",
      "I\t29801\t29980\toutput/ATAC_Seq_peaks/All_samples_peak_3\t40\t.\t3.08132\t5.65106\t4.08756\t62\t21\t179\t179\t1.0000000\n",
      "I\t32538\t33302\toutput/ATAC_Seq_peaks/All_samples_peak_4\t94\t.\t4.20471\t11.3589\t9.42615\t444\t38\t634\t764\t0.8298429\n",
      "I\t34434\t34710\toutput/ATAC_Seq_peaks/All_samples_peak_5\t69\t.\t3.65353\t8.70275\t6.93077\t77\t30\t276\t276\t1.0000000\n",
      "I\t45291\t45875\toutput/ATAC_Seq_peaks/All_samples_peak_6\t254\t.\t8.48398\t28.3009\t25.4309\t139\t38\t555\t584\t0.9503425\n",
      "I\t58135\t58783\toutput/ATAC_Seq_peaks/All_samples_peak_7\t77\t.\t4.74139\t9.55996\t7.72537\t396\t25\t619\t648\t0.9552469\n",
      "I\t61120\t61495\toutput/ATAC_Seq_peaks/All_samples_peak_8\t29\t.\t3.0819\t4.48194\t2.99934\t282\t11\t368\t375\t0.9813333\n",
      "I\t68034\t68916\toutput/ATAC_Seq_peaks/All_samples_peak_9\t227\t.\t5.55434\t25.4522\t22.7533\t508\t119\t882\t882\t1.0000000\n",
      "I\t70956\t71510\toutput/ATAC_Seq_peaks/All_samples_peak_10\t150\t.\t4.42476\t17.2813\t15.0349\t218\t69\t554\t554\t1.0000000\n",
      "\u001b[?2004h(ATAC-Seq) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "head output/peak_coverage/SRR1822153_peaks.bed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Output description\n",
    "\n",
    "From MACS https://hbctraining.github.io/Intro-to-ChIPseq/lessons/05_peak_calling_macs.html\n",
    "\n",
    "From bedtools https://bedtools.readthedocs.io/en/latest/content/tools/coverage.html\n",
    "\n",
    "    The number of features in B that overlapped (by at least one base pair) the A interval.\n",
    "    The number of bases in A that had non-zero coverage from features in B.\n",
    "    The length of the entry in A.\n",
    "    The fraction of bases in A that had non-zero coverage from features in B.\n",
    "\n",
    "Column description:\n",
    "1. chr \n",
    "2. start \n",
    "3. end \n",
    "4. peak_ID \n",
    "5. MACS_score  \n",
    "6. MACS_strand \n",
    "7. MACS_fold_enrichment \n",
    "8. MACS_-log10(pval) \n",
    "9. MACS_-log10(qval) \n",
    "10. MACS_summit (0-based offset from chrom start)\n",
    "11. reads_in_peak \n",
    "12. peak_non-zero_bases \n",
    "13. peak_length \n",
    "14. peak_fraction_covered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
